<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Discussion &amp; Future Perspectives | Software implementations allowing new approaches toward data analysis, modeling and curation of biological knowledge for Systems Medicine</title>
  <meta name="description" content="John Zobolas’s PhD thesis, NTNU" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Discussion &amp; Future Perspectives | Software implementations allowing new approaches toward data analysis, modeling and curation of biological knowledge for Systems Medicine" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://bblodfon.github.io/my-phd-thesis/" />
  
  <meta property="og:description" content="John Zobolas’s PhD thesis, NTNU" />
  <meta name="github-repo" content="bblodfon/my-phd-thesis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Discussion &amp; Future Perspectives | Software implementations allowing new approaches toward data analysis, modeling and curation of biological knowledge for Systems Medicine" />
  
  <meta name="twitter:description" content="John Zobolas’s PhD thesis, NTNU" />
  

<meta name="author" content="John Zobolas" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="04_summary_8.html"/>
<link rel="next" href="05_links.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./index.html">John Zobolas's PhD Thesis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="01_abstract.html"><a href="01_abstract.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="02_papers.html"><a href="02_papers.html"><i class="fa fa-check"></i>Paper list</a>
<ul>
<li class="chapter" data-level="" data-path="02_papers.html"><a href="02_papers.html#primary"><i class="fa fa-check"></i>Primary</a></li>
<li class="chapter" data-level="" data-path="02_papers.html"><a href="02_papers.html#additional"><i class="fa fa-check"></i>Additional</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="03_abbrev.html"><a href="03_abbrev.html"><i class="fa fa-check"></i>Abbreviations</a></li>
<li class="chapter" data-level="" data-path="04_summary_1.html"><a href="04_summary_1.html"><i class="fa fa-check"></i>Summary</a>
<ul>
<li class="chapter" data-level="" data-path="04_summary_1.html"><a href="04_summary_1.html#one-for-all"><i class="fa fa-check"></i>One for all</a></li>
<li class="chapter" data-level="" data-path="04_summary_2.html"><a href="04_summary_2.html"><i class="fa fa-check"></i>Dots and lines</a></li>
<li class="chapter" data-level="" data-path="04_summary_3.html"><a href="04_summary_3.html"><i class="fa fa-check"></i>Knowledge from a stack of papers</a></li>
<li class="chapter" data-level="" data-path="04_summary_4.html"><a href="04_summary_4.html"><i class="fa fa-check"></i>Biological Dictionaries aid in the curation of complex knowledge</a></li>
<li class="chapter" data-level="" data-path="04_summary_5.html"><a href="04_summary_5.html"><i class="fa fa-check"></i>Sharing causal interactions with PSICQUIC</a></li>
<li class="chapter" data-level="" data-path="04_summary_6.html"><a href="04_summary_6.html"><i class="fa fa-check"></i>Biological modeling: a Prelude</a></li>
<li class="chapter" data-level="" data-path="04_summary_7.html"><a href="04_summary_7.html"><i class="fa fa-check"></i>Clean Code</a></li>
<li class="chapter" data-level="" data-path="04_summary_8.html"><a href="04_summary_8.html"><i class="fa fa-check"></i>Get the right rules</a></li>
<li class="chapter" data-level="" data-path="04_summary_9.html"><a href="04_summary_9.html"><i class="fa fa-check"></i>Discussion &amp; Future Perspectives</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html"><i class="fa fa-check"></i>Links to software, documentation and data analyses</a>
<ul>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#github-org-links"><i class="fa fa-check"></i>GitHub organizations</a></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#doc-links"><i class="fa fa-check"></i>Documentation</a></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#druglogics-soft-links"><i class="fa fa-check"></i>DrugLogics software modules</a></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#r-community-packages"><i class="fa fa-check"></i>R community packages</a></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#misc-links"><i class="fa fa-check"></i>Miscellaneous data analyses and repositories</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="06_refs.html"><a href="06_refs.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Software implementations allowing new approaches toward data analysis, modeling and curation of biological knowledge for Systems Medicine</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="discussion-future-perspectives" class="section level2 unnumbered">
<h2>Discussion &amp; Future Perspectives</h2>
<div class="quote">
<blockquote style='border:none'>&ldquo;The case for my life, [...] is this: that I have added something to knowledge, and helped others to add more; and that these somethings have a value which differs in degree only, and not in kind, from that of the creations of the great mathematicians, or of any of the other artists, great or small, who have left some kind of memorial behind them.&rdquo; </br><strong>G.H. Hardy, A Mathematician's Apology</strong> </blockquote>
</div>
<p>In most of the papers included in this thesis there is a separate section discussing potential future tasks, the fulfillment of which will advance the efforts towards achieving the goals of each respective research work.
We also discussed in a previous chapter<a href="#fn16" class="footnote-ref" id="fnref16"><sup>16</sup></a> additional implementation work that is crucial for establishing a robust infrastructure for the sharing of causal molecular interactions to a wider community of users with PSICQUIC.
Here, we will discuss further implementations and research that needs to be done related to modeling efforts for combating cancer.
First, we will focus on clarifying several aspects in our own computational work and suggest future directions.
Secondly, we will reflect on the more broad problem of understanding cancer and briefly describe the challenges that the computational biology community as a whole needs to overcome to drive future modeling efforts towards enabling more proactive, predictive and personalized Systems Medicine approaches <span class="citation">[<a href="06_refs.html#ref-Hood2011" role="doc-biblioref">84</a>]</span>.</p>
<p>In <strong><a href="02_papers.html#Paper3">Paper 3</a></strong>, we outline four key prerequisites that a software pipeline designed to construct patient specific logical models should satisfy, in order to facilitate the identification of potential therapeutic solutions for cancer.
These were expressed as (1) assembling a network topology from prior knowledge databases, (2) translating baseline cancer cell line biomarker data into signaling activities, (3) calibrating logical models, created from PKNs, by modifying logical equations to match the observed signaling activities, and (4) predicting phenotypic consequences of combinatorial interventions to the simulated model behavior.
These four generic requirements were embodied in the respective software modules of Figure <a href="04_summary_7.html#fig:fig5">5</a>.
Gitsbe and Drabme were our proposed solutions for (3) and (4) and the resulting simulations and analyses was the main scope of <strong><a href="02_papers.html#Paper3">Paper 3</a></strong>.
Atopo aims to fulfill objective (1) while Aomics is the module that encapsulates the work needed for objective (2).
In the following text, we are going to discuss the reasons why these solutions were not included in <strong><a href="02_papers.html#Paper3">Paper 3</a></strong> and our future objectives for improvements on the Atopo and Aomics modules.</p>
<p>Starting with the assembly of the PKN, we previously set a list of requirements that such an output network should satisfy to be suitable for our analyses.
These were the inclusion of the drug targets so that they can be perturbed using subsequent pipeline software and the incorporation of major cancer pathways in the PKN.
Atopo is a software module (Figure <a href="04_summary_7.html#fig:fig5">5</a>) precisely implemented with these specifications in mind, using SIGNOR’s causal molecular interaction data to construct self-contained topologies that include specific signaling entities <span class="citation">[<a href="06_refs.html#ref-Flobak2016" role="doc-biblioref">104</a>]</span>.
The self-contained topology solves a practical issue, since it allows for smaller logical models with fewer fixpoint attractors, increasing computational efficiency.
It is also tightly related to the hypothesis that cancer is a disease system in itself, not dictated by external factors.
Put in other words, whatever makes a cancer cell keep proliferating, comes from within the cell itself.
Moreover, the signaling entities used in Atopo, refer in our case not only to the drug targets, but also to nodes that define the phenotypic output of the computational models later in the pipeline.
In the end, due to the methodology used to prune the network in Atopo and the molecular interaction content included in SIGNOR’s database (dated April 2018), the Atopo-generated PKN was hard to make sense of, especially when trying to interpret the simulation results in terms of mechanistic insights.
For example, some analyses using the <code>emba</code> <code>R</code> package from <strong><a href="02_papers.html#Paper4">Paper 4</a></strong> produced confusing results with regard to the nodes designated as important for the manifestation of particular synergies.<a href="#fn17" class="footnote-ref" id="fnref17"><sup>17</sup></a>
Since we had curated the CASCADE family of topologies from literature to incorporate cancer signaling pathways and associated key regulatory targets, it presented itself as a trustable and already refined solution that perfectly fitted our needs.
Our future goal is to combine molecular interaction data from different causal knowledge databases, using software like PSICQUIC <span class="citation">[<a href="06_refs.html#ref-Aranda2011" role="doc-biblioref">61</a>]</span> or OmniPath <span class="citation">[<a href="06_refs.html#ref-Turei2016" role="doc-biblioref">142</a>]</span>, to build larger and more comprehensive PKNs, suitable for our logical modeling applications in the context of cancer signaling.</p>
<p>To derive an accurate activity profile for the signaling entities in CASCADE, we used multiple omics datasets such as copy number alterations (CNA) and expression data (transcriptomics or proteomics) pertaining to each of the cell lines of interest in the reference drug combination dataset chosen for our simulations <span class="citation">[<a href="06_refs.html#ref-Flobak2019" role="doc-biblioref">116</a>]</span>.
These datasets were given as input to appropriate software tools to predict the signaling activity information <span class="citation">[<a href="06_refs.html#ref-Vaske2010" role="doc-biblioref">143</a>]</span>.
Extensive effort was spent in subsequent software, which resulted in the Aomics family of internal tools (Figure <a href="04_summary_7.html#fig:fig5">5</a>).
Sadly, we failed to produce reliable signaling activities for the 8 cell lines of the reference drug combination dataset.
For example, model output nodes that are known to be inactivated in cancer cells (e.g. the family of Caspases), were computed as being active, contradicting basic biological knowledge.
The conversion of various omics datasets to binary signaling information and their efficient integration for modeling purposes is an emerging area of research <span class="citation">[<a href="06_refs.html#ref-Beal2018" role="doc-biblioref">85</a>,<a href="06_refs.html#ref-Martignetti2016" role="doc-biblioref">144</a>–<a href="06_refs.html#ref-Dugourd2021" role="doc-biblioref">146</a>]</span>.
We acknowledge the fact that such discretization methodologies might be a bit coarse and ill-suited given the continuous nature of some omics datasets, but they are nonetheless a prerequisite for qualitative modeling methodologies such as ours.
We hope that in the future more standardized methods will be available to handle such datasets and make them better suited for our needs.</p>
<p>Overall, the Aomics endeavor represents a practical example where validation of the various input data sources of a scientific software fails and so, as researchers, we have to use every means available at our disposal to further progress our goals (i.e. in our case this meant to use a curated PKN and literature-derived signaling observations).
Nonetheless, we can exploit the information presented in such input data sources to further investigate their quality, which can reveal the varying degrees of importance with which such inputs affect the simulation results of the software.
For example, in <strong><a href="02_papers.html#Paper3">Paper 3</a></strong> we investigated how severely the pipeline’s synergy prediction performance was affected by either changes to the input training activity data or the input topology (PKN).
In the end, the results indicated that the topology is far more important, demonstrating the need for high quality prior knowledge and the significance of related curation efforts that translate scientific literature to structured knowledge infrastructures.</p>
<hr />
<p>The work described in this thesis focuses on the qualitative modeling of a single cancer cell, aiming at predicting its phenotypic behavior subject to various drug perturbations.
Our work, along with similar efforts from the Computational Biology community, has only been the first step in the mechanistic modeling of biological systems.
To help scientists better understand cancer cell biology, we need to achieve a better understanding of the processes occurring inside the cell and pave the way for the systematic analysis of cells on a broader scale than what is currently possible.
Up to now, researchers have been constructing, simulating and analyzing models to answer specific questions pertaining to context-specific modeling scenarios.
Such concentrated modeling efforts are usually restrictive, because of their limited scope and the choice of a single mathematical framework to formalize the respective models (e.g. Boolean mathematics or ODEs).
Thus the resulting biological models are inadequate and can not provide the necessary temporal and spatial resolution of the modeled systems that is required to holistically describe and interpret complex cellular behavior.</p>
<p>There is a gradual shift to replace focused models with larger, multiscale hybrid models.
These types of fine-grained models can incorporate multiple levels of granularity in their associated mathematical representation and simulated molecular scale, and their aim is to provide an accurate description of the cellular phenotype from its respective genotype <span class="citation">[<a href="06_refs.html#ref-Karr2012" role="doc-biblioref">147</a>]</span>.
Such in-silico whole-cell (WC) models are powerful scientific tools that will allow us to identify gaps in our understanding of cell biology and unify our fragmented knowledge of disease development.
Their main advantage is that they can be used to address multiple scientific questions, and conduct complex in-silico experiments that would otherwise be impractical to perform in the laboratory with current technologies.
Such approaches to modeling complex biological systems are foreseen to have a significant impact in a number of applications, both in research and industry (e.g. biotechnology), serving as a platform to facilitate model-driven discovery <span class="citation">[<a href="06_refs.html#ref-Carrera2015" role="doc-biblioref">148</a>]</span>.
In parallel, research on multicellular modeling aims to improve our understanding of the interactions between the different cells that send and receive signals to communicate and perform tissue- and organism-level functions.
Such models can help us describe in more detail the processes that drive tumor pathogenesis and favor the development and progression of cancer <span class="citation">[<a href="06_refs.html#ref-Senft2017" role="doc-biblioref">149</a>]</span>.
For example, multicellular models could elucidate the mechanisms by which tumor associated macrophages (TAMs) interact with cancer cells in the inflammatory microenvironment of solid tumors <span class="citation">[<a href="06_refs.html#ref-Fukuda2012" role="doc-biblioref">150</a>,<a href="06_refs.html#ref-Marku2020" role="doc-biblioref">151</a>]</span>, provide comprehensive explanations of how multicellular cancer resistance manifests <span class="citation">[<a href="06_refs.html#ref-Desoize2000" role="doc-biblioref">152</a>]</span>, and propose novel therapeutic targets that damage the communication of tumor cells with their microenvironment <span class="citation">[<a href="06_refs.html#ref-Komohara2016" role="doc-biblioref">153</a>]</span>.</p>
<p>There exist several challenges in building accurate and comprehensive WC models <span class="citation">[<a href="06_refs.html#ref-whole-cell-doc" role="doc-biblioref">154</a>]</span>.
To fully characterize the function of every gene product and predict the dynamics of all molecular species of a cell over its entire life cycle, WC models need to be able to synthesize information that is subject to different molecular as well as spatiotemporal scales, and perform multi-algorithmic simulations.
Despite the fact that several software tools have been built to enable the construction and simulation of WC models <span class="citation">[<a href="06_refs.html#ref-Tomita1999" role="doc-biblioref">155</a>,<a href="06_refs.html#ref-Blinov2017" role="doc-biblioref">156</a>]</span>, a complete integration of all the different algorithmic methodologies representing molecular detail at every possible scale, is currently lacking.
Moreover, a proper theoretical foundation that specifies how the different modeling formalisms can work under a hybrid mathematical framework for WC modeling is also missing <span class="citation">[<a href="06_refs.html#ref-Karr2015" role="doc-biblioref">157</a>]</span>.
On another front, the heterogeneous data needed for the scalable design, construction, calibration, simulation and validation of WC models, are incomplete, imprecise and noisy, calling for the development of new experimental methods and tools to more properly characterize cells at multiple levels of molecular granularity and expression (e.g. different omics data) <span class="citation">[<a href="06_refs.html#ref-Macklin2020" role="doc-biblioref">158</a>]</span>.
Literature curation tools, used to annotate and extract knowledge from scientific publications, complement the aforementioned efforts by providing a systematic way to assemble the comprehensive prior knowledge that is required for WC modeling.
Therefore, software implementations allowing for new approaches towards the annotation and sharing of causal molecular interactions (as were presented in this thesis), significantly contribute to such efforts.
In addition, we anticipate that the use of machine learning and text mining automated tools are going to be detrimental for future curation efforts, leveraging the vast amounts of biological knowledge for the creation of more accurate WC models.</p>
<p>There are major computational bottlenecks that we need to overcome to enable the comprehensive modeling of complex cells.
Due to the high computational costs required for the calibration and simulation of WC models (e.g. parameter estimation is particularly resource intensive), high-performance modeling algorithms need to be implemented, along with better parallel processing simulators <span class="citation">[<a href="06_refs.html#ref-Hallock2014" role="doc-biblioref">159</a>]</span>.
Additionally, technological advancements such as cloud and high performance computing (HPC) services <span class="citation">[<a href="06_refs.html#ref-HPC-wiki" role="doc-biblioref">160</a>]</span> will allow us to take advantage of modern computational resources and harness their scalability and processing power, surpassing the limitations of conventional single machines that are unsuitable for such large-scale modeling efforts.
On another front, WC models and their subcomponents (e.g. signaling pathway models corresponding to distinct cellular processes) need to be interoperable to allow for proper model integration, extension and reuse, as well as enable reproducible simulation results.
During the 2015 WC Modeling Summer School event, efforts directed towards translating a WC model to community formats such as SBML <span class="citation">[<a href="06_refs.html#ref-Keating2020" role="doc-biblioref">161</a>]</span> (for model encoding) and SBGN <span class="citation">[<a href="06_refs.html#ref-Novere2009" role="doc-biblioref">45</a>]</span> (for model visualization), indicated the need for additional standards, databases and software to accelerate WC modeling <span class="citation">[<a href="06_refs.html#ref-Waltemath2016" role="doc-biblioref">162</a>]</span>.</p>
<p>Current approaches to build comprehensive WC models of simple organisms such as the pathogen <em>Mycoplasma genitalium</em> <span class="citation">[<a href="06_refs.html#ref-Karr2012" role="doc-biblioref">147</a>,<a href="06_refs.html#ref-Burke2020" role="doc-biblioref">163</a>]</span>, and the bacterium <em>Escherichia Coli</em> <span class="citation">[<a href="06_refs.html#ref-Carrera2014" role="doc-biblioref">164</a>]</span> constitute significant achievements towards addressing all the aforementioned challenges.
Moreover, they pave the way for the construction of mammalian WC models and eventually whole-organism models <span class="citation">[<a href="06_refs.html#ref-Szigeti2014" role="doc-biblioref">165</a>,<a href="06_refs.html#ref-Viceconti2016" role="doc-biblioref">166</a>]</span> (the pinnacle of which should be a complete human model), where the communication and coordination between different WC models (representing different cell types), at different hierarchical levels, is going to be one of the most difficult issues to tackle.
Despite the difficulty inherent in such aspiring projects, many recent research efforts have been focused on studying multicellular systems with potential applications in cancer immunology and therapeutics.
Notably, several tools have been developed using agent-based or similar modeling approaches, providing ways to intuitively represent multicellular biological systems <span class="citation">[<a href="06_refs.html#ref-Ghaffarizadeh2018" role="doc-biblioref">167</a>–<a href="06_refs.html#ref-Stoll2020" role="doc-biblioref">171</a>]</span>.
Such methodologies facilitate the integration of multiple scales for the study of cell population dynamics, and some also incorporate various spatial aspects of the modeled systems.
The respective software simulators make use of coarse-grained characterizations of the interacting cells (i.e. the agents), significantly reducing the computational simulation costs.
Even though current multicellular models do not encapsulate a realistic picture of the intracellular signaling (since it is impractical to incorporate full WC models) nor of the cell communication mechanics, they have been successfully used to validate several experimental findings and study interacting cells in dynamic tissue microenvironments such as heterogeneous cell tumors.
Moreover, multicellular models have the potential to assist in the exploration of the effects that the genetic alteration of individual cells has to the population level (e.g. how knocking out genes in specific cells can limit tumor growth), the investigation of response to various anti-cancer treatments, as well as the study of the cellular mechanisms and dynamics of carcinogenesis.
We can only expect that in the future, biological and medical applications will push the boundaries of what is achievable by such software modeling solutions and that multicellular and WC models will drive progress in the cancer-related research areas and beyond.</p>
<p>Lastly, we stress the necessity of interdisciplinarity and collaboration across the research and industrial spectrum, to solve all the grand challenges involved in the modeling of complex biological systems and the efforts to combat diseases such as cancer.
Building a cohesive Computational Biology community also plays an important role, as it promotes a <em>common vision</em> and a collaborative spirit amongst the members of the community.
The success story of the COMBINE initiative’s standardization efforts in computational biological modeling, has been empowered by the promotion of its standards via tutorials, workshops and dedicated sessions at international conferences <span class="citation">[<a href="06_refs.html#ref-Hucka2015" role="doc-biblioref">172</a>]</span>.
As scientists, we are called to take risks, not only to explore the unknown and yet uncharted territory of our world but also to face the social and communication challenges between ourselves.
Only together, as one community, can we set the world in order, tackle the problems of today, and create a better tomorrow for all humankind.</p>
<div class="quote">
<blockquote style="border:none">&ldquo;Be the change you want to see in the world.&rdquo; </br><strong> Gandhi </strong></blockquote>
</div>

</div>
</div>



<div class="footnotes">
<hr />
<ol start="16">
<li id="fn16"><p>See <a href="04_summary_5.html#psicquic-chapter">Sharing causal interactions with PSICQUIC</a>.<a href="04_summary_9.html#fnref16" class="footnote-back">↩︎</a></p></li>
<li id="fn17"><p>See <code>gitsbe-model-analysis</code> repository <a href="05_links.html#misc-links">here</a><a href="04_summary_9.html#fnref17" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="04_summary_8.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="05_links.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/bblodfon/my-phd-thesis/blob/main/04_summary_9.Rmd",
"text": null
},
"download": [["phd-thesis.pdf", "PDF"]],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
