<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Sharing causal interactions with PSICQUIC | Software implementations allowing new approaches toward data analysis, modeling and curation of biological knowledge for Systems Medicine</title>
  <meta name="description" content="John Zobolas’s PhD thesis, NTNU" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Sharing causal interactions with PSICQUIC | Software implementations allowing new approaches toward data analysis, modeling and curation of biological knowledge for Systems Medicine" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://bblodfon.github.io/my-phd-thesis/" />
  
  <meta property="og:description" content="John Zobolas’s PhD thesis, NTNU" />
  <meta name="github-repo" content="bblodfon/my-phd-thesis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Sharing causal interactions with PSICQUIC | Software implementations allowing new approaches toward data analysis, modeling and curation of biological knowledge for Systems Medicine" />
  
  <meta name="twitter:description" content="John Zobolas’s PhD thesis, NTNU" />
  

<meta name="author" content="John Zobolas" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="04_summary_4.html"/>
<link rel="next" href="04_summary_6.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./index.html">John Zobolas's PhD Thesis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="01_abstract.html"><a href="01_abstract.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="02_papers.html"><a href="02_papers.html"><i class="fa fa-check"></i>Paper list</a>
<ul>
<li class="chapter" data-level="" data-path="02_papers.html"><a href="02_papers.html#primary"><i class="fa fa-check"></i>Primary</a></li>
<li class="chapter" data-level="" data-path="02_papers.html"><a href="02_papers.html#additional"><i class="fa fa-check"></i>Additional</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="03_abbrev.html"><a href="03_abbrev.html"><i class="fa fa-check"></i>Abbreviations</a></li>
<li class="chapter" data-level="" data-path="04_summary_1.html"><a href="04_summary_1.html"><i class="fa fa-check"></i>Summary</a>
<ul>
<li class="chapter" data-level="" data-path="04_summary_1.html"><a href="04_summary_1.html#one-for-all"><i class="fa fa-check"></i>One for all</a></li>
<li class="chapter" data-level="" data-path="04_summary_2.html"><a href="04_summary_2.html"><i class="fa fa-check"></i>Dots and lines</a></li>
<li class="chapter" data-level="" data-path="04_summary_3.html"><a href="04_summary_3.html"><i class="fa fa-check"></i>Knowledge from a stack of papers</a></li>
<li class="chapter" data-level="" data-path="04_summary_4.html"><a href="04_summary_4.html"><i class="fa fa-check"></i>Biological Dictionaries aid in the curation of complex knowledge</a></li>
<li class="chapter" data-level="" data-path="04_summary_5.html"><a href="04_summary_5.html"><i class="fa fa-check"></i>Sharing causal interactions with PSICQUIC</a></li>
<li class="chapter" data-level="" data-path="04_summary_6.html"><a href="04_summary_6.html"><i class="fa fa-check"></i>Biological modeling: a Prelude</a></li>
<li class="chapter" data-level="" data-path="04_summary_7.html"><a href="04_summary_7.html"><i class="fa fa-check"></i>Clean Code</a></li>
<li class="chapter" data-level="" data-path="04_summary_8.html"><a href="04_summary_8.html"><i class="fa fa-check"></i>Get the right rules</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html"><i class="fa fa-check"></i>Links to software, documentation and data analyses</a>
<ul>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#github-org-links"><i class="fa fa-check"></i>GitHub organizations</a></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#doc-links"><i class="fa fa-check"></i>Documentation</a></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#druglogics-soft-links"><i class="fa fa-check"></i>DrugLogics software modules</a></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#r-community-packages"><i class="fa fa-check"></i>R community packages</a></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#misc-links"><i class="fa fa-check"></i>Miscellaneous data analyses and repositories</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="06_refs.html"><a href="06_refs.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Software implementations allowing new approaches toward data analysis, modeling and curation of biological knowledge for Systems Medicine</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sharing-causal-interactions-with-psicquic" class="section level2 unnumbered">
<h2>Sharing causal interactions with PSICQUIC</h2>
<blockquote>&ldquo;Knowledge shared = knowledge<sup>2</sup>.&rdquo; ~ Anonymous</blockquote>
<p>Defining standards is a very important initiative across scientific disciplines, since it facilitates the accessibility and sharing of information amongst data users as well as the interoperability of software tools that produce or process the respective data, thereby increasing the quality of research findings <span class="citation">[<a href="06_refs.html#ref-Wilkinson2016" role="doc-biblioref">54</a>]</span>.
Following this logic, after the curation of causal molecular interactions from scientific literature has been achieved, the next step is to store this information to a standard data format.
One of the most detailed, community-standard formats for representing molecular interaction data is the PSI molecular interaction (MI) XML format, released by the Human Proteome Organization Proteomics Standards Initiative (HUPO-PSI) <span class="citation">[<a href="06_refs.html#ref-Hermjakob2004" role="doc-biblioref">57</a>]</span>.
The newest version of this standard is the PSI-MI XML 3.0 <span class="citation">[<a href="06_refs.html#ref-Sivade2018" role="doc-biblioref">58</a>]</span>.
A simplified format for interaction data was provided by the same organization, called the Molecular Interaction Tabular format (PSI-MITAB).
PSI-MITAB has become popular amongst the scientific community since it is more user-friendly and Microsoft Excel-compatible, compared to the respective XML-based format <span class="citation">[<a href="06_refs.html#ref-Kerrien2007" role="doc-biblioref">59</a>]</span>.
PSI-MITAB version 2.7 in particular, encapsulated many details of interest regarding a molecular interaction in a total of 42 columns, but it did not include information about its causality.
This resulted in an effort to standardize the signaling information and subsequent vocabulary terminology for causal molecular interactions, originally led by SIGNOR’s database curators <span class="citation">[<a href="06_refs.html#ref-Licata2019" role="doc-biblioref">60</a>]</span>.
The new PSI-MITAB version 2.8 (also called <strong>CausalTAB</strong>), included four new columns to incorporate additional details related to a molecular interaction’s directionality (defining the biological roles of the regulator and target, one column each), regulatory mechanism (e.g. indirect causal regulation or post-translational modification) and resulting effect (up or down regulation of the target) <span class="citation">[<a href="06_refs.html#ref-Perfetto2019" role="doc-biblioref">7</a>]</span>.</p>
<p>After having a signaling data format for molecular interactions in place, the next challenge was to find a way to share such information efficiently.
The heart of the problem was the same even without the addition of causality information: a large number of molecular interaction databases exist, each one with different APIs to access the respective data.
Since no single database can incorporate the totality of molecular interactions pertaining to a specific biological system of interest, users have to collect data from diverse databases by launching queries in different websites or by directly downloading the respective files, which might not always be offered in standardized formats.
To ease the computational access and retrieval of molecular interaction data from various resources, a web service with a common query interface (PSICQUIC) and language (MIQL) was developed <span class="citation">[<a href="06_refs.html#ref-Aranda2011" role="doc-biblioref">61</a>]</span>.
Using the PSICQUIC web service, users can now download all relevant data files from their databases of interest in different PSI-MI compliant formats, suitable for further analysis (Figure <a href="04_summary_5.html#fig:fig4">4</a>).</p>

<div style="page-break-after: always;"></div>
<div class="figure"><span id="fig:fig4"></span>
<img src="img/psicquic.png" alt="PSICQUIC architecture. Molecular interaction knowledge about a biological system, supported by different experimental methods, is being reported in publications. Each of these publications reports part of the actual truth about the studied system. This knowledge is curated from the respective publications and inserted to diverse molecular interaction databases. The databases share their data in standard formats (e.g. PSI-MITAB) via the PSICQUIC web service and are part of a registry list. Users launch queries via a PSICQUIC web client to retrieve the distributed molecular interaction data and synthesize the complete observed knowledge of the studied system, suitable for further analysis and visualization." width="90%" />
<p class="caption">
Figure 4: PSICQUIC architecture. Molecular interaction knowledge about a biological system, supported by different experimental methods, is being reported in publications. Each of these publications reports part of the actual truth about the studied system. This knowledge is curated from the respective publications and inserted to diverse molecular interaction databases. The databases share their data in standard formats (e.g. PSI-MITAB) via the PSICQUIC web service and are part of a registry list. Users launch queries via a PSICQUIC web client to retrieve the distributed molecular interaction data and synthesize the complete observed knowledge of the studied system, suitable for further analysis and visualization.
</p>
</div>
<p>With a new signaling data format established by the relevant scientific community and the PSICQUIC web service contributing to the accessibility of molecular interaction data, the next step was to support the PSI-MITAB 2.8 format in PSICQUIC.
A software project to extend the PSICQUIC platform and include causality information of molecular interactions was thus formed.<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>
Our contribution to this effort was the development of the underlying PSICQUIC software (version 1.4) that indexes CausalTAB files provided by the respective data providers, enabling the query and subsequent download of causality-enriched interactions.
The PSICQUIC View website source code was updated to show the four new columns in the HTML table results.
Additionally, several clients that enable programmatic access to the distributed signaling data (written in Java, Python and Perl programming languages), were refactored to comply with data fitting the new standard format.
Lastly, the relevant PSICQUIC documentation was improved and reformatted to enhance user readability <span class="citation">[<a href="06_refs.html#ref-mitab28-doc" role="doc-biblioref">62</a>]</span>.</p>
<p>The aforementioned development effort spurred a series of actions that led to several improvements in the PSICQUIC platform.
The Molecular Interactions Community, is an open source community providing tools, standard formats, ontologies and modules for manipulating molecular interaction data <span class="citation">[<a href="06_refs.html#ref-mi-github" role="doc-biblioref">63</a>]</span>.
For example, some of these modules are used to read and write PSI-MITAB files across different versions.
Since these modules hadn’t been updated for years (showing signs of software rot <span class="citation">[<a href="06_refs.html#ref-wiki-soft-rot" role="doc-biblioref">64</a>]</span>), we had to refactor the codebase and add tests to ensure it’s future reliability and quality.
In the end, even though we managed to complete the task and support CausalTAB in PSICQUIC via updating these modules, the need to replace them with a newer library was imminent.
JAMI is such a library, integrating all standard molecular interaction data formats such as the PSI-MI XML and PSI-MITAB, into a unified implementation, hiding the complex details of each format from the developers and thereby making their work easier <span class="citation">[<a href="06_refs.html#ref-Sivade2018a" role="doc-biblioref">65</a>]</span>.
The implementation work was initiated in a GREEKC workshop <span class="citation">[<a href="06_refs.html#ref-Hinxton2018" role="doc-biblioref">66</a>]</span> and continued during the first ELIXIR BioHackathon in Paris <span class="citation">[<a href="06_refs.html#ref-biohack2018" role="doc-biblioref">67</a>]</span>.
Upon finishing the support for CausalTAB in JAMI,<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a> we provided the first PSICQUIC service indexing SIGNOR’s CausalTAB data at the time, made available through a development server <span class="citation">[<a href="06_refs.html#ref-psicquic-causalTAB-dataset" role="doc-biblioref">68</a>]</span>.</p>
<p>During the ELIXIR BioHackathon, the architecture details of a new cloud-based, distributed PSICQUIC service were discussed and documented for future development efforts.
The goal is to enable the data providers to upload their molecular interaction data in a fully automated process and add support for data validation.
This service will minimize the long-term commitment and maintenance from the data providers, which they cannot always afford (e.g. deployment of the server hosting PSICQUIC).
Another outcome of the BioHackathon was the draft implementation of a new PSICQUIC View interface, aiming to modernize and update the current web application used to access PSICQUIC <span class="citation">[<a href="06_refs.html#ref-psicquic-view" role="doc-biblioref">69</a>]</span>.
Further work needs to be done to import and use newer technologies in the interface, which will result in better filtering and sorting of the HTML table results and more interactive, graph-based visualizations of the PSICQUIC data.
To broadly facilitate the sharing of causal interaction data, additional development efforts are needed, in particular towards improving existing PSICQUIC clients.
One such example is the PSICQUIC Universal Client, a Cytoscape app for querying multiple PSICQUIC-compliant interaction data services from a simple user interface <span class="citation">[<a href="06_refs.html#ref-PSICQUICUniversalClient" role="doc-biblioref">70</a>]</span>.
This client has been used in tutorials to guide novice users in the visualization and analysis of molecular interaction networks <span class="citation">[<a href="06_refs.html#ref-Millan2013" role="doc-biblioref">71</a>]</span>.
Lastly, two more clients that need to be updated to support the latest CausalTAB signaling format are the PSICQUIC <span class="citation">[<a href="06_refs.html#ref-Shannon2020" role="doc-biblioref">72</a>]</span> and PItools <code>R</code> packages <span class="citation">[<a href="06_refs.html#ref-Kleshchevnikov2021" role="doc-biblioref">73</a>]</span>.
These packages enable the translation of molecular interaction data directly into formats suitable for computational analysis with <code>R</code>, and therefore are crucial for relevant computational tasks.</p>
<div style="page-break-after: always;"></div>

</div>
<div class="footnotes">
<hr />
<ol start="3">
<li id="fn3"><p>This project was funded by the GREEKC (Gene Regulation Ensemble Effort for the Knowledge Commons) COST action (<a href="https://www.greekc.org/" class="uri">https://www.greekc.org/</a>) and was realized as a Short Term Scientific Mission (STSM) in cooperation with engineers from the IntAct team at the European Bioinformatics Institute (EBI).<a href="04_summary_5.html#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p>During the GREEKC Marseille Hackathon 2019 event, see more info on the project here: <a href="https://github.com/GREEKC/hackathon-marseille/tree/master/project_descriptions/causal_psicquic" class="uri">https://github.com/GREEKC/hackathon-marseille/tree/master/project_descriptions/causal_psicquic</a><a href="04_summary_5.html#fnref4" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="04_summary_4.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="04_summary_6.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/bblodfon/my-phd-thesis/blob/main/04_summary_5.Rmd",
"text": null
},
"download": [["phd-thesis.pdf", "PDF"]],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
