<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | Software implementations allowing new approaches toward data analysis, modeling and curation of biological knowledge for Systems Medicine</title>
  <meta name="description" content="John Zobolas’s PhD thesis, NTNU" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="References | Software implementations allowing new approaches toward data analysis, modeling and curation of biological knowledge for Systems Medicine" />
  <meta property="og:type" content="book" />
  <meta property="og:url" content="https://bblodfon.github.io/my-phd-thesis/" />
  
  <meta property="og:description" content="John Zobolas’s PhD thesis, NTNU" />
  <meta name="github-repo" content="bblodfon/my-phd-thesis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | Software implementations allowing new approaches toward data analysis, modeling and curation of biological knowledge for Systems Medicine" />
  
  <meta name="twitter:description" content="John Zobolas’s PhD thesis, NTNU" />
  

<meta name="author" content="John Zobolas" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="05_links.html"/>

<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>



<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./index.html">John Zobolas's PhD Thesis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Acknowledgements</a></li>
<li class="chapter" data-level="" data-path="01_abstract.html"><a href="01_abstract.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="02_papers.html"><a href="02_papers.html"><i class="fa fa-check"></i>Paper list</a>
<ul>
<li class="chapter" data-level="" data-path="02_papers.html"><a href="02_papers.html#primary"><i class="fa fa-check"></i>Primary</a></li>
<li class="chapter" data-level="" data-path="02_papers.html"><a href="02_papers.html#additional"><i class="fa fa-check"></i>Additional</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="03_abbrev.html"><a href="03_abbrev.html"><i class="fa fa-check"></i>Abbreviations</a></li>
<li class="chapter" data-level="" data-path="04_summary_1.html"><a href="04_summary_1.html"><i class="fa fa-check"></i>Summary</a>
<ul>
<li class="chapter" data-level="" data-path="04_summary_1.html"><a href="04_summary_1.html#one-for-all"><i class="fa fa-check"></i>One for all</a></li>
<li class="chapter" data-level="" data-path="04_summary_2.html"><a href="04_summary_2.html"><i class="fa fa-check"></i>Dots and lines</a></li>
<li class="chapter" data-level="" data-path="04_summary_3.html"><a href="04_summary_3.html"><i class="fa fa-check"></i>Knowledge from a stack of papers</a></li>
<li class="chapter" data-level="" data-path="04_summary_4.html"><a href="04_summary_4.html"><i class="fa fa-check"></i>Biological Dictionaries aid in the curation of complex knowledge</a></li>
<li class="chapter" data-level="" data-path="04_summary_5.html"><a href="04_summary_5.html"><i class="fa fa-check"></i>Sharing causal interactions with PSICQUIC</a></li>
<li class="chapter" data-level="" data-path="04_summary_6.html"><a href="04_summary_6.html"><i class="fa fa-check"></i>Biological modeling: a Prelude</a></li>
<li class="chapter" data-level="" data-path="04_summary_7.html"><a href="04_summary_7.html"><i class="fa fa-check"></i>Clean Code</a></li>
<li class="chapter" data-level="" data-path="04_summary_8.html"><a href="04_summary_8.html"><i class="fa fa-check"></i>Get the right rules</a></li>
<li class="chapter" data-level="" data-path="04_summary_9.html"><a href="04_summary_9.html"><i class="fa fa-check"></i>Discussion &amp; Future Perspectives</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html"><i class="fa fa-check"></i>Links to software, documentation and data analyses</a>
<ul>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#github-org-links"><i class="fa fa-check"></i>GitHub organizations</a></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#doc-links"><i class="fa fa-check"></i>Documentation</a></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#druglogics-soft-links"><i class="fa fa-check"></i>DrugLogics software modules</a></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#r-community-packages"><i class="fa fa-check"></i>R community packages</a></li>
<li class="chapter" data-level="" data-path="05_links.html"><a href="05_links.html#misc-links"><i class="fa fa-check"></i>Miscellaneous data analyses and repositories</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="06_refs.html"><a href="06_refs.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Software implementations allowing new approaches toward data analysis, modeling and curation of biological knowledge for Systems Medicine</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered">
<h1>References</h1>

<div id="refs" class="references csl-bib-body hanging-indent" line-spacing="2">
<div id="ref-UBDs" class="csl-entry">
1. Zobolas, J., Touré, V., Kuiper, M., &amp; Vercruysse, S. (2020). <span class="nocase">UniBioDicts: Unified access to Biological Dictionaries</span>. <em>Bioinformatics</em>, <em>37</em>(1), 143–144. <a href="https://doi.org/10.1093/bioinformatics/btaa1065">https://doi.org/10.1093/bioinformatics/btaa1065</a>
</div>
<div id="ref-Zobolas2020-pubdict" class="csl-entry">
2. Zobolas, J., Kim, J.-D., Kuiper, M., &amp; Vercruysse, S. (2020). <em><span class="nocase">Linking PubDictionaries with UniBioDicts to support Community Curation</span></em>. BioHackrXiv. <a href="https://doi.org/10.37044/osf.io/gzfa8">https://doi.org/10.37044/osf.io/gzfa8</a>
</div>
<div id="ref-Zobolas2020" class="csl-entry">
3. Zobolas, J., Kuiper, M., &amp; Flobak, Å. (2020). <span class="nocase">emba: R package for analysis and visualization of biomarkers in boolean model ensembles</span>. <em>Journal of Open Source Software</em>, <em>5</em>(53), 2583. <a href="https://doi.org/10.21105/joss.02583">https://doi.org/10.21105/joss.02583</a>
</div>
<div id="ref-Zobolas2021-bias" class="csl-entry">
4. Zobolas, J., Monteiro, P. T., Kuiper, M., &amp; Flobak, Å. (2021). <em><span class="nocase">Boolean function metrics can assist modelers to check and choose logical rules</span></em>. <a href="http://arxiv.org/abs/2104.01279">http://arxiv.org/abs/2104.01279</a>
</div>
<div id="ref-vsm-box" class="csl-entry">
5. Vercruysse, S., Zobolas, J., Touré, V., Andersen, M. K., &amp; Kuiper, M. (2020). <span class="nocase">VSM-box: general-purpose interface for biocuration and knowledge representation</span>. <em>Preprints</em>. <a href="https://doi.org/10.20944/preprints202007.0557.v1">https://doi.org/10.20944/preprints202007.0557.v1</a>
</div>
<div id="ref-Toure2021" class="csl-entry">
6. Touré, V., Zobolas, J., Kuiper, M., &amp; Vercruysse, S. (2021). <span class="nocase">CausalBuilder: bringing the MI2CAST causal interaction annotation standard to the curator</span>. <em>Database</em>. <a href="https://doi.org/10.1093/database/baaa107">https://doi.org/10.1093/database/baaa107</a>
</div>
<div id="ref-Perfetto2019" class="csl-entry">
7. Perfetto, L., Acencio, M. L., Bradley, G., Cesareni, G., Del Toro, N., Fazekas, D., Hermjakob, H., Korcsmaros, T., Kuiper, M., Lægreid, A., Lo Surdo, P., Lovering, R. C., Orchard, S., Porras, P., Thomas, P. D., Touré, V., Zobolas, J., &amp; Licata, L. (2019). <span class="nocase">CausalTAB: the PSI-MITAB 2.8 updated format for signalling data representation and dissemination</span>. <em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/btz132">https://doi.org/10.1093/bioinformatics/btz132</a>
</div>
<div id="ref-Abbate2000" class="csl-entry">
8. Abbate, J. (2000). <em><span class="nocase">Inventing the internet</span></em>. MIT press.
</div>
<div id="ref-Naughton2016" class="csl-entry">
9. Naughton, J. (2016). <span class="nocase">The evolution of the Internet: from military experiment to General Purpose Technology</span>. <em>Journal of Cyber Policy</em>, <em>1</em>(1), 5–28. <a href="https://doi.org/10.1080/23738871.2016.1157619">https://doi.org/10.1080/23738871.2016.1157619</a>
</div>
<div id="ref-Polasky2019" class="csl-entry">
10. Polasky, S., Kling, C. L., Levin, S. A., Carpenter, S. R., Daily, G. C., Ehrlich, P. R., Heal, G. M., &amp; Lubchenco, J. (2019). <span class="nocase">Role of economics in analyzing the environment and sustainable development</span>. <em>Proceedings of the National Academy of Sciences of the United States of America</em>, <em>116</em>(12), 5233–5238. <a href="https://doi.org/10.1073/pnas.1901616116">https://doi.org/10.1073/pnas.1901616116</a>
</div>
<div id="ref-Roser2013" class="csl-entry">
11. Roser, M., Ortiz-Ospina, E., &amp; Ritchie, H. (2013). <em><span>Life Expectancy</span></em>. <a href="https://ourworldindata.org/life-expectancy">https://ourworldindata.org/life-expectancy</a> (15 May 2021, date last accessed).
</div>
<div id="ref-Jinek2012" class="csl-entry">
12. Jinek, M., Chylinski, K., Fonfara, I., Hauer, M., Doudna, J. A., &amp; Charpentier, E. (2012). <span class="nocase">A programmable dual-RNA–guided DNA endonuclease in adaptive bacterial immunity</span>. <em>Science</em>, <em>337</em>(6096), 816–821.
</div>
<div id="ref-Zhu2020" class="csl-entry">
13. Zhu, H., Li, C., &amp; Gao, C. (2020). <span class="nocase">Applications of CRISPR–Cas in agriculture and plant biotechnology</span>. <em>Nature Reviews Molecular Cell Biology</em>, <em>21</em>(11), 661–677. <a href="https://doi.org/10.1038/s41580-020-00288-9">https://doi.org/10.1038/s41580-020-00288-9</a>
</div>
<div id="ref-Bailey2020" class="csl-entry">
14. Bailey, R., &amp; Tupy, M. L. (2020). <em><span>Ten Global Trends Every Smart Person Should Know: And Many Others You Will Find Interesting</span></em>. Cato Institute.
</div>
<div id="ref-Gibbons1999" class="csl-entry">
15. Gibbons, M. (1999). <span class="nocase">Science’s new social contract with society</span>. <em>Nature</em>, <em>402</em>, C81–C84. <a href="https://doi.org/10.1038/35011576">https://doi.org/10.1038/35011576</a>
</div>
<div id="ref-HAQ2015" class="csl-entry">
16. <em><span class="nocase">Healthcare Access and Quality Index</span></em>. (2015). <a href="https://ourworldindata.org/grapher/healthcare-access-and-quality-index">https://ourworldindata.org/grapher/healthcare-access-and-quality-index</a> (15 May 2021, date last accessed).
</div>
<div id="ref-Apweiler2018" class="csl-entry">
17. Apweiler, R., Beissbarth, T., Berthold, M. R., Blüthgen, N., Burmeister, Y., Dammann, O., Deutsch, A., Feuerhake, F., Franke, A., Hasenauer, J., Hoffmann, S., Höfer, T., Jansen, P. L., Kaderali, L., Klingmüller, U., Koch, I., Kohlbacher, O., Kuepfer, L., Lammert, F., … Wolkenhauer, O. (2018). <span class="nocase">Whither systems medicine?</span> <em>Experimental <span>&amp;</span> Molecular Medicine</em>, <em>50</em>(3), e453. <a href="https://doi.org/10.1038/emm.2017.290">https://doi.org/10.1038/emm.2017.290</a>
</div>
<div id="ref-Trudeau1976" class="csl-entry">
18. Trudeau, R. J. (1976). <em><span class="nocase">Dots and lines</span></em>. Kent State University Press.
</div>
<div id="ref-Erdos1960" class="csl-entry">
19. Erdos, P., &amp; Rényi, A. (1960). <span class="nocase">On the evolution of random graphs</span>. <em>Publ. Math. Inst. Hung. Acad. Sci</em>, <em>5</em>(1), 17–60.
</div>
<div id="ref-Barabasi1999" class="csl-entry">
20. Barabási, A. L., &amp; Albert, R. (1999). <span class="nocase">Emergence of scaling in random networks</span>. <em>Science</em>, <em>286</em>(5439), 509–512. <a href="https://doi.org/10.1126/science.286.5439.509">https://doi.org/10.1126/science.286.5439.509</a>
</div>
<div id="ref-Barabasi2013" class="csl-entry">
21. Barabási, A. L. (2013). <span class="nocase">Network science</span>. <em>Philosophical Transactions of the Royal Society A: Mathematical, Physical and Engineering Sciences</em>, <em>371</em>(1987). <a href="https://doi.org/10.1098/rsta.2012.0375">https://doi.org/10.1098/rsta.2012.0375</a>
</div>
<div id="ref-Csardi2006" class="csl-entry">
22. Csardi, G., &amp; Nepusz, T. (2006). <span class="nocase">The igraph software package for complex network research</span>. <em>InterJournal, Complex Systems</em>, <em>1695</em>(5), 1–9. <a href="http://igraph.org">http://igraph.org</a>
</div>
<div id="ref-Bastian2009" class="csl-entry">
23. Bastian, M., Heymann, S., &amp; Jacomy, M. (2009). <span class="nocase">Gephi: an open source software for exploring and manipulating networks</span>. <em>Proceedings of the International AAAI Conference on Web and Social Media</em>.
</div>
<div id="ref-Mrvar2016" class="csl-entry">
24. Mrvar, A., &amp; Batagelj, V. (2016). <span class="nocase">Analysis and visualization of large networks with program package Pajek</span>. <em>Complex Adaptive Systems Modeling</em>, <em>4</em>(1), 1–8. <a href="https://doi.org/10.1186/s40294-016-0017-8">https://doi.org/10.1186/s40294-016-0017-8</a>
</div>
<div id="ref-Shannon2003" class="csl-entry">
25. Shannon, P., Markiel, A., Ozier, O., Baliga, N. S., Wang, J. T., Ramage, D., Amin, N., Schwikowski, B., &amp; Ideker, T. (2003). <span class="nocase">Cytoscape: a software environment for integrated models of biomolecular interaction networks</span>. <em>Genome Research</em>, <em>13</em>(11), 2498–2504.
</div>
<div id="ref-Dahlquist2002" class="csl-entry">
26. Dahlquist, K. D., Salomonis, N., Vranizan, K., Lawlor, S. C., &amp; Conklin, B. R. (2002). <span class="nocase">GenMAPP, a new tool for viewing and analyzing microarray data on biological pathways</span>. <em>Nature Genetics</em>, <em>31</em>(1), 19–20. <a href="https://doi.org/10.1038/ng0502-19">https://doi.org/10.1038/ng0502-19</a>
</div>
<div id="ref-Breitkreutz2003" class="csl-entry">
27. Breitkreutz, B.-J., Stark, C., &amp; Tyers, M. (2003). <span class="nocase">Osprey: a network visualization system</span>. <em>Genome Biology</em>, <em>4</em>(3), R22. <a href="https://doi.org/10.1186/gb-2003-4-3-r22">https://doi.org/10.1186/gb-2003-4-3-r22</a>
</div>
<div id="ref-Funahashi2003" class="csl-entry">
28. Funahashi, A., Morohashi, M., Kitano, H., &amp; Tanimura, N. (2003). <span class="nocase">CellDesigner: a process diagram editor for gene-regulatory and biochemical networks</span>. <em>BIOSILICO</em>, <em>1</em>(5), 159–162. <a href="https://doi.org/10.1016/s1478-5382(03)02370-9">https://doi.org/10.1016/s1478-5382(03)02370-9</a>
</div>
<div id="ref-Sidiropoulos2017" class="csl-entry">
29. Sidiropoulos, K., Viteri, G., Sevilla, C., Jupe, S., Webber, M., Orlic-Milacic, M., Jassal, B., May, B., Shamovsky, V., Duenas, C., Rothfels, K., Matthews, L., Song, H., Stein, L., Haw, R., D’Eustachio, P., Ping, P., Hermjakob, H., &amp; Fabregat, A. (2017). <span class="nocase">Reactome enhanced pathway visualization</span>. <em>Bioinformatics</em>, <em>33</em>(21), 3461–3467. <a href="https://doi.org/10.1093/bioinformatics/btx441">https://doi.org/10.1093/bioinformatics/btx441</a>
</div>
<div id="ref-Smith2009" class="csl-entry">
30. Smith, M. A., Shneiderman, B., Milic-Frayling, N., Mendes Rodrigues, E., Barash, V., Dunne, C., Capone, T., Perer, A., &amp; Gleave, E. (2009). <span class="nocase">Analyzing (social media) networks with NodeXL</span>. <em>Proceedings of the Fourth International Conference on Communities and Technologies - c<span>&amp;</span>t ’09</em>, 255. <a href="https://doi.org/10.1145/1556460.1556497">https://doi.org/10.1145/1556460.1556497</a>
</div>
<div id="ref-Kalamaras2014" class="csl-entry">
31. Kalamaras, D. (2014). <em><span class="nocase">Social Networks Visualizer (SocNetV): Social network analysis and visualization software</span></em>. <a href="http://socnetv.org">http://socnetv.org</a>
</div>
<div id="ref-Chatterjee2021" class="csl-entry">
32. Chatterjee, T., Albert, R., Thapliyal, S., Azarhooshang, N., &amp; DasGupta, B. (2021). <span class="nocase">Detecting network anomalies using Forman–Ricci curvature and a case study for human brain networks</span>. <em>Scientific Reports</em>, <em>11</em>(1), 8121. <a href="https://doi.org/10.1038/s41598-021-87587-z">https://doi.org/10.1038/s41598-021-87587-z</a>
</div>
<div id="ref-Mislove2007" class="csl-entry">
33. Mislove, A., Marcon, M., Gummadi, K. P., Druschel, P., &amp; Bhattacharjee, B. (2007). <span class="nocase">Measurement and analysis of online social networks</span>. <em>Proceedings of the ACM SIGCOMM Internet Measurement Conference, IMC</em>, 29–42. <a href="https://doi.org/10.1145/1298306.1298311">https://doi.org/10.1145/1298306.1298311</a>
</div>
<div id="ref-Maheshwari2020" class="csl-entry">
34. Maheshwari, P., &amp; Albert, R. (2020). <span class="nocase">Network model and analysis of the spread of Covid-19 with social distancing</span>. <em>Applied Network Science</em>, <em>5</em>(1), 1–13. <a href="https://doi.org/10.1007/s41109-020-00344-5">https://doi.org/10.1007/s41109-020-00344-5</a>
</div>
<div id="ref-Barabasi2011" class="csl-entry">
35. Barabási, A.-L., Gulbahce, N., &amp; Loscalzo, J. (2011). <span class="nocase">Network medicine: a network-based approach to human disease.</span> <em>Nature Reviews. Genetics</em>, <em>12</em>(1), 56–68. <a href="https://doi.org/10.1038/nrg2918">https://doi.org/10.1038/nrg2918</a>
</div>
<div id="ref-Vercruysse2019a" class="csl-entry">
36. Vercruysse, S. (2019). <em><span>VSM Pages</span></em>. <a href="https://vsm.github.io/vsm-pages/intro">https://vsm.github.io/vsm-pages/intro</a> (15 May 2021, date last accessed).
</div>
<div id="ref-Howe2008" class="csl-entry">
37. Howe, D., Costanzo, M., Fey, P., Gojobori, T., Hannick, L., Hide, W., Hill, D. P., Kania, R., Schaeffer, M., St Pierre, S., Twigger, S., White, O., &amp; Yon Rhee, S. (2008). <span class="nocase">Big data: The future of biocuration</span>. <em>Nature</em>, <em>455</em>(7209), 47–50. <a href="https://doi.org/10.1038/455047a">https://doi.org/10.1038/455047a</a>
</div>
<div id="ref-Ammari2018" class="csl-entry">
38. Ammari, M., Chatr Aryamontri, A., Attrill, H., Bairoch, A., Berardini, T., Blake, J., Chen, Q., Collado, J., Dauga, D., Dudley, J. T., Engel, S., Erill, I., Fey, P., Gibson, R., Hermjakob, H., Holliday, G., Howe, D., Hunter, C., Landsman, D., … Zhang, Z. (2018). <span class="nocase">Biocuration: Distilling data into knowledge</span>. <em>PLOS Biology</em>, <em>16</em>(4), e2002846. <a href="https://doi.org/10.1371/journal.pbio.2002846">https://doi.org/10.1371/journal.pbio.2002846</a>
</div>
<div id="ref-Jenssen2001" class="csl-entry">
39. Jenssen, T.-K., Lægreid, A., Komorowski, J., &amp; Hovig, E. (2001). <span class="nocase">A literature network of human genes for high-throughput analysis of gene expression</span>. <em>Nature Genetics</em>, <em>28</em>(1), 21–28. <a href="https://doi.org/10.1038/ng0501-21">https://doi.org/10.1038/ng0501-21</a>
</div>
<div id="ref-intact-editor" class="csl-entry">
40. <em><span class="nocase">IntAct editor</span></em>. (2007). <a href="https://github.com/EBI-IntAct/intact-editor">https://github.com/EBI-IntAct/intact-editor</a> (15 May 2021, date last accessed).
</div>
<div id="ref-Rutherford2014" class="csl-entry">
41. Rutherford, K. M., Harris, M. A., Lock, A., Oliver, S. G., &amp; Wood, V. (2014). <span class="nocase">Canto: an online tool for community literature curation</span>. <em>Bioinformatics</em>, <em>30</em>(12), 1791–1792. <a href="https://doi.org/10.1093/bioinformatics/btu103">https://doi.org/10.1093/bioinformatics/btu103</a>
</div>
<div id="ref-canto-doc" class="csl-entry">
42. <em><span>Canto Documentation</span></em>. (2014). <a href="https://curation.pombase.org/pombe/docs/index/">https://curation.pombase.org/pombe/docs/index/</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-Kuperstein2013" class="csl-entry">
43. Kuperstein, I., Cohen, D. P. A., Pook, S., Viara, E., Calzone, L., Barillot, E., &amp; Zinovyev, A. (2013). <span class="nocase">NaviCell: A web-based environment for navigation, curation and maintenance of large molecular interaction maps</span>. <em>BMC Systems Biology</em>, <em>7</em>(1), 100. <a href="https://doi.org/10.1186/1752-0509-7-100">https://doi.org/10.1186/1752-0509-7-100</a>
</div>
<div id="ref-Gawron2016" class="csl-entry">
44. Gawron, P., Ostaszewski, M., Satagopam, V., Gebel, S., Mazein, A., Kuzma, M., Zorzan, S., McGee, F., Otjacques, B., Balling, R., &amp; Schneider, R. (2016). <span class="nocase">MINERVA—A platform for visualization and curation of molecular interaction networks</span>. <em>Npj Systems Biology and Applications</em>, <em>2</em>(1), 1–6. <a href="https://doi.org/10.1038/npjsba.2016.20">https://doi.org/10.1038/npjsba.2016.20</a>
</div>
<div id="ref-Novere2009" class="csl-entry">
45. Novère, N. L., Hucka, M., Mi, H., Moodie, S., Schreiber, F., Sorokin, A., Demir, E., Wegner, K., Aladjem, M. I., Wimalaratne, S. M., Bergman, F. T., Gauges, R., Ghazal, P., Kawaji, H., Li, L., Matsuoka, Y., Villéger, A., Boyd, S. E., Calzone, L., … Kitano, H. (2009). <span>The Systems Biology Graphical Notation</span>. <em>Nature Biotechnology</em>, <em>27</em>(8), 735–741. <a href="https://doi.org/10.1038/nbt.1558">https://doi.org/10.1038/nbt.1558</a>
</div>
<div id="ref-vsm-paper" class="csl-entry">
46. Vercruysse, S., &amp; Kuiper, M. (2020). <span class="nocase">Intuitive representation of computable knowledge</span>. <em>Preprints</em>. <a href="https://doi.org/10.20944/preprints202007.0486.v2">https://doi.org/10.20944/preprints202007.0486.v2</a>
</div>
<div id="ref-Toure2020" class="csl-entry">
47. Touré, V., Vercruysse, S., Acencio, M. L., Lovering, R., Orchard, S., Bradley, G., Casals-Casas, C., Chaouiya, C., Del-Toro, N., Flobak, Å., Gaudet, P., Hermjakob, H., Licata, L., Lægreid, A., Mungall, C., Niknejad, A., Panni, S., Perfetto, L., Porras, P., … Kuiper, M. (2020). <span class="nocase">The Minimum Information about a Molecular Interaction Causal Statement (MI2CAST)</span>. <em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/btaa622">https://doi.org/10.1093/bioinformatics/btaa622</a>
</div>
<div id="ref-mi2cast-doc" class="csl-entry">
48. Vasundra, T. (2020). <em><span>MI2CAST Documentation</span></em>. <a href="https://github.com/MI2CAST/MI2CAST/blob/master/docs/MI2CAST_guideline.md">https://github.com/MI2CAST/MI2CAST/blob/master/docs/MI2CAST_guideline.md</a> (15 May 2021, date last accessed).
</div>
<div id="ref-TheUniProtConsortium2019" class="csl-entry">
49. The UniProt Consortium. (2019). <span class="nocase">UniProt: a worldwide hub of protein knowledge</span>. <em>Nucleic Acids Research</em>, <em>47</em>(D1), D506–D515. <a href="https://doi.org/10.1093/nar/gky1049">https://doi.org/10.1093/nar/gky1049</a>
</div>
<div id="ref-Meldal2019" class="csl-entry">
50. Meldal, B. H. M., Bye-A-Jee, H., Gajdoš, L., Hammerová, Z., Horáčková, A., Melicher, F., Perfetto, L., Pokorný, D., Lopez, M. R., Türková, A., Wong, E. D., Xie, Z., Casanova, E. B., Del-Toro, N., Koch, M., Porras, P., Hermjakob, H., &amp; Orchard, S. (2019). <span class="nocase">Complex Portal 2018: extended content and enhanced visualization tools for macromolecular complexes</span>. <em>Nucleic Acids Research</em>, <em>47</em>(D1), D550–D558. <a href="https://doi.org/10.1093/nar/gky1001">https://doi.org/10.1093/nar/gky1001</a>
</div>
<div id="ref-RNACentral2018" class="csl-entry">
51. The RNAcentral Consortium. (2018). <span class="nocase">RNAcentral: a hub of information for non-coding RNA sequences</span>. <em>Nucleic Acids Research</em>, <em>47</em>(D1), D1250–D1251. <a href="https://doi.org/10.1093/nar/gky1206">https://doi.org/10.1093/nar/gky1206</a>
</div>
<div id="ref-Whetzel2011" class="csl-entry">
52. Whetzel, P. L., Noy, N. F., Shah, N. H., Alexander, P. R., Nyulas, C., Tudorache, T., &amp; Musen, M. A. (2011). <span class="nocase">BioPortal: enhanced functionality via new Web services from the National Center for Biomedical Ontology to access and use ontologies in software applications.</span> <em>Nucleic Acids Research</em>, <em>39</em>(Web Server issue), W541–5. <a href="https://doi.org/10.1093/nar/gkr469">https://doi.org/10.1093/nar/gkr469</a>
</div>
<div id="ref-Madeira2019" class="csl-entry">
53. Madeira, F., Park, Y. mi, Lee, J., Buso, N., Gur, T., Madhusoodanan, N., Basutkar, P., Tivey, A. R. N., Potter, S. C., Finn, R. D., &amp; Lopez, R. (2019). <span class="nocase">The EMBL-EBI search and sequence analysis tools APIs in 2019</span>. <em>Nucleic Acids Research</em>, <em>47</em>(W1), W636–W641. <a href="https://doi.org/10.1093/nar/gkz268">https://doi.org/10.1093/nar/gkz268</a>
</div>
<div id="ref-Wilkinson2016" class="csl-entry">
54. Wilkinson, M. D., Dumontier, M., Aalbersberg, Ij. J., Appleton, G., Axton, M., Baak, A., Blomberg, N., Boiten, J. W., da Silva Santos, L. B., Bourne, P. E., Bouwman, J., Brookes, A. J., Clark, T., Crosas, M., Dillo, I., Dumon, O., Edmunds, S., Evelo, C. T., Finkers, R., … Mons, B. (2016). <span class="nocase">The FAIR Guiding Principles for scientific data management and stewardship</span>. <em>Scientific Data</em>, <em>3</em>(1), 1–9. <a href="https://doi.org/10.1038/sdata.2016.18">https://doi.org/10.1038/sdata.2016.18</a>
</div>
<div id="ref-Hartmann2019" class="csl-entry">
55. Hartmann, N. B., Hüffer, T., Thompson, R. C., Hassellöv, M., Verschoor, A., Daugaard, A. E., Rist, S., Karlsson, T., Brennholt, N., Cole, M., Herrling, M. P., Hess, M. C., Ivleva, N. P., Lusher, A. L., &amp; Wagner, M. (2019). <span class="nocase">Are We Speaking the Same Language? Recommendations for a Definition and Categorization Framework for Plastic Debris</span>. <em>Environmental Science and Technology</em>, <em>53</em>(3), 1039–1047. <a href="https://doi.org/10.1021/acs.est.8b05297">https://doi.org/10.1021/acs.est.8b05297</a>
</div>
<div id="ref-Kim2019" class="csl-entry">
56. Kim, J.-D., Wang, Y., Fujiwara, T., Okuda, S., Callahan, T. J., &amp; Cohen, K. B. (2019). <span class="nocase">Open Agile text mining for bioinformatics: the PubAnnotation ecosystem</span>. <em>Bioinformatics</em>, <em>35</em>(21), 4372–4380. <a href="https://doi.org/10.1093/bioinformatics/btz227">https://doi.org/10.1093/bioinformatics/btz227</a>
</div>
<div id="ref-Hermjakob2004" class="csl-entry">
57. Hermjakob, H., Montecchi-Palazzi, L., Bader, G., Wojcik, J., Salwinski, L., Ceol, A., Moore, S., Orchard, S., Sarkans, U., Mering, C. von, Roechert, B., Poux, S., Jung, E., Mersch, H., Kersey, P., Lappe, M., Li, Y., Zeng, R., Rana, D., … Apweiler, R. (2004). <span class="nocase">The HUPO PSI’s Molecular Interaction format—a community standard for the representation of protein interaction data</span>. <em>Nature Biotechnology</em>, <em>22</em>(2), 177–183. <a href="https://doi.org/10.1038/nbt926">https://doi.org/10.1038/nbt926</a>
</div>
<div id="ref-Sivade2018" class="csl-entry">
58. Sivade, M., Alonso-López, D., Ammari, M., Bradley, G., Campbell, N. H., Ceol, A., Cesareni, G., Combe, C., De Las Rivas, J., Del-Toro, N., Heimbach, J., Hermjakob, H., Jurisica, I., Koch, M., Licata, L., Lovering, R. C., Lynn, D. J., Meldal, B. H. M., Micklem, G., … Orchard, S. (2018). <span class="nocase">Encompassing new use cases - level 3.0 of the HUPO-PSI format for molecular interactions</span>. <em>BMC Bioinformatics</em>, <em>19</em>(1), 134. <a href="https://doi.org/10.1186/s12859-018-2118-1">https://doi.org/10.1186/s12859-018-2118-1</a>
</div>
<div id="ref-Kerrien2007" class="csl-entry">
59. Kerrien, S., Orchard, S., Montecchi-Palazzi, L., Aranda, B., Quinn, A. F., Vinod, N., Bader, G. D., Xenarios, I., Wojcik, J., Sherman, D., Tyers, M., Salama, J. J., Moore, S., Ceol, A., Chatr-aryamontri, A., Oesterheld, M., Stümpflen, V., Salwinski, L., Nerothin, J., … Hermjakob, H. (2007). <span class="nocase">Broadening the horizon – level 2.5 of the HUPO-PSI format for molecular interactions</span>. <em>BMC Biology</em>, <em>5</em>(1), 44. <a href="https://doi.org/10.1186/1741-7007-5-44">https://doi.org/10.1186/1741-7007-5-44</a>
</div>
<div id="ref-Licata2019" class="csl-entry">
60. Licata, L., Lo Surdo, P., Iannuccelli, M., Palma, A., Micarelli, E., Perfetto, L., Peluso, D., Calderone, A., Castagnoli, L., &amp; Cesareni, G. (2019). <span class="nocase">SIGNOR 2.0, the SIGnaling Network Open Resource 2.0: 2019 update</span>. <em>Nucleic Acids Research</em>. <a href="https://doi.org/10.1093/nar/gkz949">https://doi.org/10.1093/nar/gkz949</a>
</div>
<div id="ref-Aranda2011" class="csl-entry">
61. Aranda, B., Blankenburg, H., Kerrien, S., Brinkman, F. S. L., Ceol, A., Chautard, E., Dana, J. M., De Las Rivas, J., Dumousseau, M., Galeota, E., Gaulton, A., Goll, J., Hancock, R. E. W., Isserlin, R., Jimenez, R. C., Kerssemakers, J., Khadake, J., Lynn, D. J., Michaut, M., … Hermjakob, H. (2011). <span class="nocase">PSICQUIC and PSISCORE: accessing and scoring molecular interactions</span>. <em>Nature Methods</em>, <em>8</em>(7), 528–529. <a href="https://doi.org/10.1038/nmeth.1637">https://doi.org/10.1038/nmeth.1637</a>
</div>
<div id="ref-mitab28-doc" class="csl-entry">
62. <em><span>MITAB 2.8 Documentation</span></em>. (2018). <a href="http://psicquic.github.io/MITAB28Format.html">http://psicquic.github.io/MITAB28Format.html</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-mi-github" class="csl-entry">
63. <em><span>Molecular Interactions Community</span></em>. (2015). <a href="https://github.com/MICommunity">https://github.com/MICommunity</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-soft-rot-wiki" class="csl-entry">
64. <em><span class="nocase">Software rot (Wikipedia entry)</span></em>. (2014). <a href="https://en.wikipedia.org/wiki/Software_rot">https://en.wikipedia.org/wiki/Software_rot</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-Sivade2018a" class="csl-entry">
65. Sivade, M., Koch, M., Shrivastava, A., Alonso-López, D., De Las Rivas, J., Del-Toro, N., Combe, C. W., Meldal, B. H. M., Heimbach, J., Rappsilber, J., Sullivan, J., Yehudi, Y., &amp; Orchard, S. (2018). <span class="nocase">JAMI: a Java library for molecular interactions and data interoperability</span>. <em>BMC Bioinformatics</em>, <em>19</em>(1), 133. <a href="https://doi.org/10.1186/s12859-018-2119-0">https://doi.org/10.1186/s12859-018-2119-0</a>
</div>
<div id="ref-Hinxton2018" class="csl-entry">
66. <em><span>GREEKC Hinxton Workshop</span></em>. (2018). <a href="https://www.greekc.org/activity/hinxton-workshop/">https://www.greekc.org/activity/hinxton-workshop/</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-biohack2018" class="csl-entry">
67. <em><span>ELIXIR BioHackathon Paris</span></em>. (2018). <a href="https://2018.biohackathon-europe.org/">https://2018.biohackathon-europe.org/</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-psicquic-causalTAB-dataset" class="csl-entry">
68. Del-Toro, N., Zobolas, J., &amp; Touré, V. (2019). <em><span class="nocase">Signor Dataset in CausalTAB (PSICQUIC Dev Server)</span></em>. <a href="http://wwwdev.ebi.ac.uk/Tools/webservices/psicquic/causality/webservices/current/search/query/*?firstResult=0&amp;maxResults=10&amp;format=tab28">http://wwwdev.ebi.ac.uk/Tools/webservices/psicquic/causality/webservices/current/search/query/*?firstResult=0&amp;maxResults=10&amp;format=tab28</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-psicquic-view" class="csl-entry">
69. <em><span>PSICQUIC View</span></em>. (2011). <a href="http://www.ebi.ac.uk/Tools/webservices/psicquic/view/main.xhtml">http://www.ebi.ac.uk/Tools/webservices/psicquic/view/main.xhtml</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-PSICQUICUniversalClient" class="csl-entry">
70. <em><span class="nocase">PSICQUIC Universal Client (Cytoscape app)</span></em>. (2012). <a href="https://apps.cytoscape.org/apps/psicquicuniversalclient">https://apps.cytoscape.org/apps/psicquicuniversalclient</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-Millan2013" class="csl-entry">
71. Millán, P. P. (2013). <span class="nocase">Visualization and analysis of biological networks</span>. <em>Methods in Molecular Biology</em>, <em>1021</em>, 63–88. <a href="https://doi.org/10.1007/978-1-62703-450-0_4">https://doi.org/10.1007/978-1-62703-450-0_4</a>
</div>
<div id="ref-Shannon2020" class="csl-entry">
72. Shannon, P. (2020). <em><span class="nocase">PSICQUIC: Proteomics Standard Initiative Common QUery InterfaCe. R package version 1.28.0.</span></em> <a href="https://doi.org/10.18129/B9.bioc.PSICQUIC">https://doi.org/10.18129/B9.bioc.PSICQUIC</a>
</div>
<div id="ref-Kleshchevnikov2021" class="csl-entry">
73. Kleshchevnikov, V. (2021). <em><span class="nocase">PItools: Protein interaction data tools. R package</span></em>. GitHub. <a href="https://github.com/vitkl/PItools">https://github.com/vitkl/PItools</a>
</div>
<div id="ref-Wang2012" class="csl-entry">
74. Wang, R.-S., Saadatpour, A., &amp; Albert, R. (2012). <span class="nocase">Boolean modeling in systems biology: an overview of methodology and applications</span>. <em>Physical Biology</em>, <em>9</em>(5), 55001. <a href="https://doi.org/10.1088/1478-3975/9/5/055001">https://doi.org/10.1088/1478-3975/9/5/055001</a>
</div>
<div id="ref-Aldridge2006" class="csl-entry">
75. Aldridge, B. B., Burke, J. M., Lauffenburger, D. A., &amp; Sorger, P. K. (2006). <span class="nocase">Physicochemical modelling of cell signalling pathways</span>. <em>Nature Cell Biology</em>, <em>8</em>(11), 1195–1203. <a href="https://doi.org/10.1038/ncb1497">https://doi.org/10.1038/ncb1497</a>
</div>
<div id="ref-ElKalaawy2015" class="csl-entry">
76. ElKalaawy, N., &amp; Wassal, A. (2015). <span class="nocase">Methodologies for the modeling and simulation of biochemical networks, illustrated for signal transduction pathways: A primer</span>. <em>Biosystems</em>, <em>129</em>, 1–18. <a href="https://doi.org/10.1016/J.BIOSYSTEMS.2015.01.008">https://doi.org/10.1016/J.BIOSYSTEMS.2015.01.008</a>
</div>
<div id="ref-Groen2021" class="csl-entry">
77. Groen, D., Arabnejad, H., Jancauskas, V., Edeling, W. N., Jansson, F., Richardson, R. A., Lakhlili, J., Veen, L., Bosak, B., Kopta, P., Wright, D. W., Monnier, N., Karlshoefer, P., Suleimenova, D., Sinclair, R., Vassaux, M., Nikishova, A., Bieniek, M., Luk, O. O., … Coveney, P. V. (2021). <span class="nocase">VECMAtk: a scalable verification, validation and uncertainty quantification toolkit for scientific simulations</span>. <em>Philosophical Transactions of the Royal Society A: Mathematical, Physical and Engineering Sciences</em>, <em>379</em>(2197). <a href="https://doi.org/10.1098/rsta.2020.0221">https://doi.org/10.1098/rsta.2020.0221</a>
</div>
<div id="ref-Schwab2020" class="csl-entry">
78. Schwab, J. D., Kühlwein, S. D., Ikonomi, N., Kühl, M., &amp; Kestler, H. A. (2020). <span class="nocase">Concepts in Boolean network modeling: What do they all mean?</span> <em>Computational and Structural Biotechnology Journal</em>, <em>18</em>, 571–582. <a href="https://doi.org/10.1016/j.csbj.2020.03.001">https://doi.org/10.1016/j.csbj.2020.03.001</a>
</div>
<div id="ref-Kauffman1969" class="csl-entry">
79. Kauffman, S. A. (1969). <span class="nocase">Metabolic stability and epigenesis in randomly constructed genetic nets</span>. <em>Journal of Theoretical Biology</em>, <em>22</em>(3), 437–467. <a href="https://doi.org/10.1016/0022-5193(69)90015-0">https://doi.org/10.1016/0022-5193(69)90015-0</a>
</div>
<div id="ref-Kauffman1993" class="csl-entry">
80. Kauffman, S. A. (1993). <em><span class="nocase">The origins of order: Self-organization and selection in evolution</span></em>. Oxford University Press, USA.
</div>
<div id="ref-Faure2006" class="csl-entry">
81. Faure, A., Naldi, A., Chaouiya, C., &amp; Thieffry, D. (2006). <span class="nocase">Dynamical analysis of a generic Boolean model for the control of the mammalian cell cycle</span>. <em>Bioinformatics</em>, <em>22</em>(14), e124–e131. <a href="https://doi.org/10.1093/bioinformatics/btl210">https://doi.org/10.1093/bioinformatics/btl210</a>
</div>
<div id="ref-Abou-Jaoude2016" class="csl-entry">
82. Abou-Jaoudé, W., Traynard, P., Monteiro, P. T., Saez-Rodriguez, J., Helikar, T., Thieffry, D., &amp; Chaouiya, C. (2016). <span class="nocase">Logical Modeling and Dynamical Analysis of Cellular Networks</span>. <em>Frontiers in Genetics</em>, <em>7</em>, 94. <a href="https://doi.org/10.3389/fgene.2016.00094">https://doi.org/10.3389/fgene.2016.00094</a>
</div>
<div id="ref-Naldi2018a" class="csl-entry">
83. Naldi, A., Hernandez, C., Levy, N., Stoll, G., Monteiro, P. T., Chaouiya, C., Helikar, T., Zinovyev, A., Calzone, L., Cohen-Boulakia, S., Thieffry, D., &amp; Paulevé, L. (2018). <span class="nocase">The CoLoMoTo Interactive Notebook: Accessible and Reproducible Computational Analyses for Qualitative Biological Networks</span>. <em>Frontiers in Physiology</em>, <em>9</em>, 680. <a href="https://doi.org/10.3389/fphys.2018.00680">https://doi.org/10.3389/fphys.2018.00680</a>
</div>
<div id="ref-Hood2011" class="csl-entry">
84. Hood, L., &amp; Friend, S. H. (2011). <span class="nocase">Predictive, personalized, preventive, participatory (P4) cancer medicine</span>. <em>Nature Reviews Clinical Oncology</em>, <em>8</em>(3), 184–187. <a href="https://doi.org/10.1038/nrclinonc.2010.227">https://doi.org/10.1038/nrclinonc.2010.227</a>
</div>
<div id="ref-Beal2018" class="csl-entry">
85. Béal, J., Montagud, A., Traynard, P., Barillot, E., &amp; Calzone, L. (2018). <span class="nocase">Personalization of Logical Models With Multi-Omics Data Allows Clinical Stratification of Patients</span>. <em>Frontiers in Physiology</em>, <em>9</em>, 1965. <a href="https://doi.org/10.3389/fphys.2018.01965">https://doi.org/10.3389/fphys.2018.01965</a>
</div>
<div id="ref-Eduati2017" class="csl-entry">
86. Eduati, F., Doldàn-Martelli, V., Klinger, B., Cokelaer, T., Sieber, A., Kogera, F., Dorel, M., Garnett, M. J., Blüthgen, N., &amp; Saez-Rodriguez, J. (2017). <span class="nocase">Drug Resistance Mechanisms in Colorectal Cancer Dissected with Cell Type-Specific Dynamic Logic Models.</span> <em>Cancer Research</em>, <em>77</em>(12), 3364–3375. <a href="https://doi.org/10.1158/0008-5472.CAN-17-0078">https://doi.org/10.1158/0008-5472.CAN-17-0078</a>
</div>
<div id="ref-Beal2021" class="csl-entry">
87. Béal, J., Pantolini, L., Noël, V., Barillot, E., &amp; Calzone, L. (2021). <span class="nocase">Personalized logical models to investigate cancer response to BRAF treatments in melanomas and colorectal cancers</span>. <em>PLOS Computational Biology</em>, <em>17</em>(1), e1007900. <a href="https://doi.org/10.1371/journal.pcbi.1007900">https://doi.org/10.1371/journal.pcbi.1007900</a>
</div>
<div id="ref-Tognetti2021" class="csl-entry">
88. Tognetti, M., Gabor, A., Yang, M., Cappelletti, V., Windhager, J., Rueda, O. M., Charmpi, K., Esmaeilishirazifard, E., Bruna, A., Souza, N. de, Caldas, C., Beyer, A., Picotti, P., Saez-Rodriguez, J., &amp; Bodenmiller, B. (2021). <span class="nocase">Deciphering the signaling network of breast cancer improves drug sensitivity prediction</span>. <em>Cell Systems</em>, <em>12</em>. <a href="https://doi.org/10.1016/j.cels.2021.04.002">https://doi.org/10.1016/j.cels.2021.04.002</a>
</div>
<div id="ref-Saadatpour2011" class="csl-entry">
89. Saadatpour, A., Wang, R.-S., Liao, A., Liu, X., Loughran, T. P., Albert, I., &amp; Albert, R. (2011). <span class="nocase">Dynamical and Structural Analysis of a T Cell Survival Network Identifies Novel Candidate Therapeutic Targets for Large Granular Lymphocyte Leukemia</span>. <em>PLoS Computational Biology</em>, <em>7</em>(11), e1002267. <a href="https://doi.org/10.1371/journal.pcbi.1002267">https://doi.org/10.1371/journal.pcbi.1002267</a>
</div>
<div id="ref-Flobak2015" class="csl-entry">
90. Flobak, Å., Baudot, A., Remy, E., Thommesen, L., Thieffry, D., Kuiper, M., &amp; Lægreid, A. (2015). <span class="nocase">Discovery of Drug Synergies in Gastric Cancer Cells Predicted by Logical Modeling</span>. <em>PLOS Computational Biology</em>, <em>11</em>(8). <a href="https://doi.org/10.1371/journal.pcbi.1004426">https://doi.org/10.1371/journal.pcbi.1004426</a>
</div>
<div id="ref-Eduati2020" class="csl-entry">
91. Eduati, F., Jaaks, P., Wappler, J., Cramer, T., Merten, C. A., Garnett, M. J., &amp; Saez‐Rodriguez, J. (2020). <span class="nocase">Patient‐specific logic models of signaling pathways from screenings on cancer biopsies to prioritize personalized combination therapies</span>. <em>Molecular Systems Biology</em>, <em>16</em>(2). <a href="https://doi.org/10.15252/msb.20188664">https://doi.org/10.15252/msb.20188664</a>
</div>
<div id="ref-Wilson2014" class="csl-entry">
92. Wilson, G., Aruliah, D. A., Brown, C. T., Chue Hong, N. P., Davis, M., Guy, R. T., Haddock, S. H. D., Huff, K. D., Mitchell, I. M., Plumbley, M. D., Waugh, B., White, E. P., &amp; Wilson, P. (2014). <span class="nocase">Best Practices for Scientific Computing</span>. <em>PLoS Biology</em>, <em>12</em>(1), e1001745. <a href="https://doi.org/10.1371/journal.pbio.1001745">https://doi.org/10.1371/journal.pbio.1001745</a>
</div>
<div id="ref-Coveney2021" class="csl-entry">
93. Coveney, P. V., &amp; Highfield, R. R. (2021). <span class="nocase">When we can trust computers (and when we can’t)</span>. <em>Philosophical Transactions. Series A, Mathematical, Physical, and Engineering Sciences</em>, <em>379</em>(2197), 20200067. <a href="https://doi.org/10.1098/rsta.2020.0067">https://doi.org/10.1098/rsta.2020.0067</a>
</div>
<div id="ref-Barnes2010" class="csl-entry">
94. Barnes, N. (2010). <span class="nocase">Publish your computer code: it is good enough</span>. <em>Nature</em>, <em>467</em>(7317), 753–753. <a href="https://doi.org/10.1038/467753a">https://doi.org/10.1038/467753a</a>
</div>
<div id="ref-Prlic2012" class="csl-entry">
95. Prlić, A., &amp; Procter, J. B. (2012). <span class="nocase">Ten Simple Rules for the Open Development of Scientific Software</span>. <em>PLoS Computational Biology</em>, <em>8</em>(12), e1002802. <a href="https://doi.org/10.1371/journal.pcbi.1002802">https://doi.org/10.1371/journal.pcbi.1002802</a>
</div>
<div id="ref-Karimzadeh2018" class="csl-entry">
96. Karimzadeh, M., &amp; Hoffman, M. M. (2018). <span class="nocase">Top considerations for creating bioinformatics software documentation</span>. <em>Briefings in Bioinformatics</em>, <em>19</em>(4), 693–699. <a href="https://doi.org/10.1093/bib/bbw134">https://doi.org/10.1093/bib/bbw134</a>
</div>
<div id="ref-List2017" class="csl-entry">
97. List, M., Ebert, P., &amp; Albrecht, F. (2017). <span class="nocase">Ten Simple Rules for Developing Usable Software in Computational Biology</span>. <em>PLoS Computational Biology</em>, <em>13</em>(1), e1005265. <a href="https://doi.org/10.1371/journal.pcbi.1005265">https://doi.org/10.1371/journal.pcbi.1005265</a>
</div>
<div id="ref-Sandve2013" class="csl-entry">
98. Sandve, G. K., Nekrutenko, A., Taylor, J., &amp; Hovig, E. (2013). <span class="nocase">Ten simple rules for reproducible computational research</span>. <em>PLoS Computational Biology</em>, <em>9</em>(10), e1003285. <a href="https://doi.org/10.1371/journal.pcbi.1003285">https://doi.org/10.1371/journal.pcbi.1003285</a>
</div>
<div id="ref-Martin2009" class="csl-entry">
99. Martin, R. C. (2009). <em><span class="nocase">Clean code: a handbook of agile software craftsmanship</span></em>. Pearson Education.
</div>
<div id="ref-gaddum1940pharmacology" class="csl-entry">
100. Gaddum, J. H. (1940). <em><span>Pharmacology</span></em>. Oxford University Press, London.
</div>
<div id="ref-Bliss1939" class="csl-entry">
101. Bliss, C. I. (1939). <span class="nocase">The Toxicity of Poisons Applied Jointly</span>. <em>Annals of Applied Biology</em>, <em>26</em>(3), 585–615. <a href="https://doi.org/10.1111/j.1744-7348.1939.tb06990.x">https://doi.org/10.1111/j.1744-7348.1939.tb06990.x</a>
</div>
<div id="ref-Galton1907" class="csl-entry">
102. Galton, F. (1907). <span class="nocase">Vox populi</span>. <em>Nature</em>, <em>75</em>(1949), 450–451. <a href="https://doi.org/10.1038/075450a0">https://doi.org/10.1038/075450a0</a>
</div>
<div id="ref-Marbach2012" class="csl-entry">
103. Marbach, D., Costello, J. C., Küffner, R., Vega, N. M., Prill, R. J., Camacho, D. M., Allison, K. R., Kellis, M., Collins, J. J., Aderhold, A., Stolovitzky, G., Bonneau, R., Chen, Y., Cordero, F., Crane, M., Dondelinger, F., Drton, M., Esposito, R., Foygel, R., … Zimmer, R. (2012). <span class="nocase">Wisdom of crowds for robust gene network inference</span>. <em>Nature Methods</em>, <em>9</em>(8), 796–804. <a href="https://doi.org/10.1038/nmeth.2016">https://doi.org/10.1038/nmeth.2016</a>
</div>
<div id="ref-Flobak2016" class="csl-entry">
104. Flobak, Å. (2016). <span class="nocase">Systems Medicine: From Modeling Systems Perturbations to Predicting Drug Synergies</span>. <em>PhD Thesis</em>. <a href="https://ntnuopen.ntnu.no/ntnu-xmlui/handle/11250/2385695">https://ntnuopen.ntnu.no/ntnu-xmlui/handle/11250/2385695</a>
</div>
<div id="ref-ApacheCommons" class="csl-entry">
105. <em><span>Apache Commons</span></em>. (2021). <a href="https://commons.apache.org/">https://commons.apache.org/</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-Maven" class="csl-entry">
106. <em><span>Apache Maven</span></em>. (2021). <a href="https://maven.apache.org/">https://maven.apache.org/</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-Naldi2018" class="csl-entry">
107. Naldi, A. (2018). <span class="nocase">BioLQM: A Java Toolkit for the Manipulation and Conversion of Logical Qualitative Models of Biological Networks</span>. <em>Frontiers in Physiology</em>, <em>9</em>, 1605. <a href="https://doi.org/10.3389/fphys.2018.01605">https://doi.org/10.3389/fphys.2018.01605</a>
</div>
<div id="ref-Naldi2018b" class="csl-entry">
108. Naldi, A., Hernandez, C., Abou-Jaoudé, W., Monteiro, P. T., Chaouiya, C., &amp; Thieffry, D. (2018). <span class="nocase">Logical Modeling and Analysis of Cellular Regulatory Networks With GINsim 3.0</span>. <em>Frontiers in Physiology</em>, <em>9</em>, 646. <a href="https://doi.org/10.3389/fphys.2018.00646">https://doi.org/10.3389/fphys.2018.00646</a>
</div>
<div id="ref-Chaouiya2013" class="csl-entry">
109. Chaouiya, C., Bérenguier, D., Keating, S. M., Naldi, A., Iersel, M. P. van, Rodriguez, N., Dräger, A., Büchel, F., Cokelaer, T., Kowal, B., Wicks, B., Gonçalves, E., Dorier, J., Page, M., Monteiro, P. T., Kamp, A. von, Xenarios, I., Jong, H. de, Hucka, M., … Helikar, T. (2013). <span class="nocase">SBML qualitative models: A model representation format and infrastructure to foster interactions between qualitative modelling formalisms and tools</span>. <em>BMC Systems Biology</em>, <em>7</em>(1), 1–15. <a href="https://doi.org/10.1186/1752-0509-7-135">https://doi.org/10.1186/1752-0509-7-135</a>
</div>
<div id="ref-Mussel2010" class="csl-entry">
110. Müssel, C., Hopfensitz, M., &amp; Kestler, H. A. (2010). <span class="nocase">BoolNet—an R package for generation, reconstruction and analysis of Boolean networks</span>. <em>Bioinformatics</em>, <em>26</em>(10), 1378–1380. <a href="https://doi.org/10.1093/bioinformatics/btq124">https://doi.org/10.1093/bioinformatics/btq124</a>
</div>
<div id="ref-Veliz-Cuba2014" class="csl-entry">
111. Veliz-Cuba, A., Aguilar, B., Hinkelmann, F., &amp; Laubenbacher, R. (2014). <span class="nocase">Steady state analysis of Boolean molecular network models via model reduction and computational algebra</span>. <em>BMC Bioinformatics</em>, <em>15</em>(1), 221. <a href="https://doi.org/10.1186/1471-2105-15-221">https://doi.org/10.1186/1471-2105-15-221</a>
</div>
<div id="ref-JUnit5" class="csl-entry">
112. <em><span>JUnit 5</span></em>. (2021). <a href="https://junit.org/junit5/">https://junit.org/junit5/</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-AssertJ" class="csl-entry">
113. <em><span class="nocase">AssertJ - fluent assertions java library</span></em>. (2021). <a href="https://assertj.github.io/doc/">https://assertj.github.io/doc/</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-Mockito-site" class="csl-entry">
114. <em><span class="nocase">Mockito framework site</span></em>. (2021). <a href="https://site.mockito.org/">https://site.mockito.org/</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-Xie2016" class="csl-entry">
115. Xie, Y. (2016). <em><span class="nocase">bookdown: Authoring Books and Technical Documents with R Markdown</span></em>. Chapman; Hall/CRC. <a href="https://bookdown.org/yihui/bookdown">https://bookdown.org/yihui/bookdown</a>
</div>
<div id="ref-Flobak2019" class="csl-entry">
116. Flobak, Å., Niederdorfer, B., Nakstad, V. T., Thommesen, L., Klinkenberg, G., &amp; Lægreid, A. (2019). <span class="nocase">A high-throughput drug combination screen of targeted small molecule inhibitors in cancer cell lines</span>. <em>Scientific Data</em>, <em>6</em>(1), 237. <a href="https://doi.org/10.1038/s41597-019-0255-7">https://doi.org/10.1038/s41597-019-0255-7</a>
</div>
<div id="ref-Flobak2017" class="csl-entry">
117. Flobak, Å., Vazquez, M., Lægreid, A., &amp; Valencia, A. (2017). <span class="nocase">CImbinator: a web-based tool for drug synergy analysis in small- and large-scale datasets</span>. <em>Bioinformatics</em>, <em>33</em>(15), 2410–2412. <a href="https://doi.org/10.1093/bioinformatics/btx161">https://doi.org/10.1093/bioinformatics/btx161</a>
</div>
<div id="ref-kegg-cancer" class="csl-entry">
118. <em><span class="nocase">KEGG PATHWAY: Pathways in cancer - Homo sapiens (human)</span></em>. (2000). <a href="https://www.genome.jp/kegg-bin/show_pathway?hsa05200">https://www.genome.jp/kegg-bin/show_pathway?hsa05200</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-Fawcett2006" class="csl-entry">
119. Fawcett, T. (2006). <span class="nocase">An introduction to ROC analysis</span>. <em>Pattern Recognition Letters</em>, <em>27</em>(8), 861–874. <a href="https://doi.org/10.1016/j.patrec.2005.10.010">https://doi.org/10.1016/j.patrec.2005.10.010</a>
</div>
<div id="ref-Saito2015" class="csl-entry">
120. Saito, T., &amp; Rehmsmeier, M. (2015). <span class="nocase">The Precision-Recall Plot Is More Informative than the ROC Plot When Evaluating Binary Classifiers on Imbalanced Datasets</span>. <em>PLOS ONE</em>, <em>10</em>(3), e0118432. <a href="https://doi.org/10.1371/journal.pone.0118432">https://doi.org/10.1371/journal.pone.0118432</a>
</div>
<div id="ref-Perkins2006" class="csl-entry">
121. Perkins, N. J., &amp; Schisterman, E. F. (2006). <span class="nocase">The Inconsistency of <span>“Optimal”</span> Cutpoints Obtained using Two Criteria based on the Receiver Operating Characteristic Curve</span>. <em>American Journal of Epidemiology</em>, <em>163</em>(7), 670–675. <a href="https://doi.org/10.1093/aje/kwj063">https://doi.org/10.1093/aje/kwj063</a>
</div>
<div id="ref-Chang2021" class="csl-entry">
122. Chang, W., Cheng, J., Allaire, J. J., Sievert, C., Schloerke, B., Xie, Y., Allen, J., McPherson, J., Dipert, A., &amp; Borges, B. (2021). <em><span class="nocase">shiny: Web Application Framework for R</span></em>. <a href="https://cran.r-project.org/package=shiny">https://cran.r-project.org/package=shiny</a>, R package version 1.6.0.
</div>
<div id="ref-Xie2021" class="csl-entry">
123. Xie, Y., Cheng, J., &amp; Tan, X. (2021). <em><span class="nocase">DT: A Wrapper of the JavaScript Library ’DataTables’</span></em>. <a href="https://cran.r-project.org/package=DT">https://cran.r-project.org/package=DT</a>, R package version 0.18.
</div>
<div id="ref-Sievert2020" class="csl-entry">
124. Sievert, C. (2020). <em><span class="nocase">Interactive Web-Based Data Visualization with R, plotly, and shiny</span></em>. Chapman; Hall/CRC. <a href="https://plotly-r.com">https://plotly-r.com</a>
</div>
<div id="ref-Grau2015" class="csl-entry">
125. Grau, J., Grosse, I., &amp; Keilwagen, J. (2015). <span class="nocase">PRROC: computing and visualizing precision-recall and receiver operating characteristic curves in R</span>. <em>Bioinformatics</em>, <em>31</em>(15), 2595–2597. <a href="https://doi.org/10.1093/bioinformatics/btv153">https://doi.org/10.1093/bioinformatics/btv153</a>
</div>
<div id="ref-Wickham2015" class="csl-entry">
126. Wickham, H. (2015). <em><span class="nocase">R packages: organize, test, document, and share your code</span></em>. O’Reilly Media, Inc.
</div>
<div id="ref-Roache1997" class="csl-entry">
127. Roache, P. J. (1997). <span class="nocase">Quantification of uncertainty in computational fluid dynamics</span>. <em>Annual Review of Fluid Mechanics</em>, <em>29</em>, 123–160. <a href="https://doi.org/10.1146/annurev.fluid.29.1.123">https://doi.org/10.1146/annurev.fluid.29.1.123</a>
</div>
<div id="ref-Mendoza2006" class="csl-entry">
128. Mendoza, L., &amp; Xenarios, I. (2006). <span class="nocase">A method for the generation of standardized qualitative dynamical systems of regulatory networks</span>. <em>Theoretical Biology and Medical Modelling</em>, <em>3</em>(1), 13. <a href="https://doi.org/10.1186/1742-4682-3-13">https://doi.org/10.1186/1742-4682-3-13</a>
</div>
<div id="ref-Faure2009" class="csl-entry">
129. Fauré, A., &amp; Thieffry, D. (2009). <span class="nocase">Logical modelling of cell cycle control in eukaryotes: A comparative study</span>. <em>Molecular BioSystems</em>, <em>5</em>(12), 1569–1581. <a href="https://doi.org/10.1039/b907562n">https://doi.org/10.1039/b907562n</a>
</div>
<div id="ref-Niederdorfer2020" class="csl-entry">
130. Niederdorfer, B., Touré, V., Vazquez, M., Thommesen, L., Kuiper, M., Lægreid, A., &amp; Flobak, Å. (2020). <span class="nocase">Strategies to Enhance Logic Modeling-Based Cell Line-Specific Drug Synergy Prediction</span>. <em>Frontiers in Physiology</em>, <em>11</em>, 862. <a href="https://doi.org/10.3389/fphys.2020.00862">https://doi.org/10.3389/fphys.2020.00862</a>
</div>
<div id="ref-Videla2016" class="csl-entry">
131. Videla, S., Saez-Rodriguez, J., Guziolowski, C., &amp; Siegel, A. (2016). Caspo: A toolbox for automated reasoning on the response of logical signaling networks families. <em>Bioinformatics</em>, <em>33</em>(6), 947–950. <a href="https://doi.org/10.1093/bioinformatics/btw738">https://doi.org/10.1093/bioinformatics/btw738</a>
</div>
<div id="ref-Gjerga2020" class="csl-entry">
132. Gjerga, E., Trairatphisan, P., Gabor, A., Koch, H., Chevalier, C., Ceccarelli, F., Dugourd, A., Mitsos, A., &amp; Saez-Rodriguez, J. (2020). <span class="nocase">Converting networks to predictive logic models from perturbation signalling data with CellNOpt</span>. <em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/btaa561">https://doi.org/10.1093/bioinformatics/btaa561</a>
</div>
<div id="ref-Aghamiri2020" class="csl-entry">
133. Aghamiri, S. S., Singh, V., Naldi, A., Helikar, T., Soliman, S., &amp; Niarakis, A. (2020). <span class="nocase">Automated inference of Boolean models from molecular interaction maps using CaSQ</span>. <em>Bioinformatics</em>. <a href="https://doi.org/10.1093/bioinformatics/btaa484">https://doi.org/10.1093/bioinformatics/btaa484</a>
</div>
<div id="ref-Helikar2012" class="csl-entry">
134. Helikar, T., Kowal, B., Madrahimov, A., Shrestha, M., Pedersen, J., Limbu, K., Thapa, I., Rowley, T., Satalkar, R., Kochi, N., Konvalina, J., &amp; Rogers, J. A. (2012). <span class="nocase">Bio-Logic Builder: A Non-Technical Tool for Building Dynamical, Qualitative Models</span>. <em>PLoS ONE</em>, <em>7</em>(10), e46417. <a href="https://doi.org/10.1371/journal.pone.0046417">https://doi.org/10.1371/journal.pone.0046417</a>
</div>
<div id="ref-Cury2019" class="csl-entry">
135. Cury, J. E. R., Monteiro, P. T., &amp; Chaouiya, C. (2019). <span class="nocase">Partial Order on the set of Boolean Regulatory Functions</span>. <em>arXiv</em>. <a href="http://arxiv.org/abs/1901.07623">http://arxiv.org/abs/1901.07623</a>
</div>
<div id="ref-Saez-Rodriguez2009" class="csl-entry">
136. Saez‐Rodriguez, J., Alexopoulos, L. G., Epperlein, J., Samaga, R., Lauffenburger, D. A., Klamt, S., &amp; Sorger, P. K. (2009). <span class="nocase">Discrete logic modelling as a means to link protein signalling networks with functional analysis of mammalian signal transduction</span>. <em>Molecular Systems Biology</em>, <em>5</em>(1), 331. <a href="https://doi.org/10.1038/msb.2009.87">https://doi.org/10.1038/msb.2009.87</a>
</div>
<div id="ref-Shmulevich2004" class="csl-entry">
137. Shmulevich, I., &amp; Kauffman, S. A. (2004). <span class="nocase">Activities and sensitivities in Boolean network models</span>. <em>Physical Review Letters</em>, <em>93</em>(4), 048701. <a href="https://doi.org/10.1103/PhysRevLett.93.048701">https://doi.org/10.1103/PhysRevLett.93.048701</a>
</div>
<div id="ref-Gherardi2016" class="csl-entry">
138. Gherardi, M., &amp; Rotondo, P. (2016). <span class="nocase">Measuring logic complexity can guide pattern discovery in empirical systems</span>. <em>Complexity</em>, <em>21</em>, 397–408. <a href="https://doi.org/10.1002/cplx.21819">https://doi.org/10.1002/cplx.21819</a>
</div>
<div id="ref-Abou-Jaoude2019" class="csl-entry">
139. Abou-Jaoudé, W., &amp; Monteiro, P. T. (2019). <span class="nocase">On logical bifurcation diagrams</span>. <em>Journal of Theoretical Biology</em>, <em>466</em>, 39–63. <a href="https://doi.org/10.1016/j.jtbi.2019.01.008">https://doi.org/10.1016/j.jtbi.2019.01.008</a>
</div>
<div id="ref-Abou-Jaoude2009" class="csl-entry">
140. Abou-Jaoudé, W., Ouattara, D. A., &amp; Kaufman, M. (2009). <span class="nocase">From structure to dynamics: Frequency tuning in the p53-Mdm2 network. I. Logical approach</span>. <em>Journal of Theoretical Biology</em>, <em>258</em>(4), 561–577. <a href="https://doi.org/10.1016/j.jtbi.2009.02.005">https://doi.org/10.1016/j.jtbi.2009.02.005</a>
</div>
<div id="ref-McInnes2018" class="csl-entry">
141. McInnes, L., Healy, J., &amp; Melville, J. (2018). <span class="nocase">UMAP: Uniform Manifold Approximation and Projection for Dimension Reduction</span>. <em>arXiv</em>. <a href="http://arxiv.org/abs/1802.03426">http://arxiv.org/abs/1802.03426</a>
</div>
<div id="ref-Turei2016" class="csl-entry">
142. Türei, D., Korcsmáros, T., &amp; Saez-Rodriguez, J. (2016). <span class="nocase">OmniPath: guidelines and gateway for literature-curated signaling pathway resources</span>. <em>Nature Methods</em>, <em>13</em>(12), 966–967. <a href="https://doi.org/10.1038/nmeth.4077">https://doi.org/10.1038/nmeth.4077</a>
</div>
<div id="ref-Vaske2010" class="csl-entry">
143. Vaske, C. J., Benz, S. C., Sanborn, J. Z., Earl, D., Szeto, C., Zhu, J., Haussler, D., &amp; Stuart, J. M. (2010). <span class="nocase">Inference of patient-specific pathway activities from multi-dimensional cancer genomics data using PARADIGM</span>. <em>Bioinformatics</em>, <em>26</em>(12), i237–i245. <a href="https://doi.org/10.1093/bioinformatics/btq182">https://doi.org/10.1093/bioinformatics/btq182</a>
</div>
<div id="ref-Martignetti2016" class="csl-entry">
144. Martignetti, L., Calzone, L., Bonnet, E., Barillot, E., &amp; Zinovyev, A. (2016). <span class="nocase">ROMA: Representation and quantification of module activity from target expression data</span>. <em>Frontiers in Genetics</em>, <em>7</em>, 18. <a href="https://doi.org/10.3389/fgene.2016.00018">https://doi.org/10.3389/fgene.2016.00018</a>
</div>
<div id="ref-Schubert2018" class="csl-entry">
145. Schubert, M., Klinger, B., Klünemann, M., Sieber, A., Uhlitz, F., Sauer, S., Garnett, M. J., Blüthgen, N., &amp; Saez-Rodriguez, J. (2018). <span class="nocase">Perturbation-response genes reveal signaling footprints in cancer gene expression</span>. <em>Nature Communications</em>, <em>9</em>(1), 1–11. <a href="https://doi.org/10.1038/s41467-017-02391-6">https://doi.org/10.1038/s41467-017-02391-6</a>
</div>
<div id="ref-Dugourd2021" class="csl-entry">
146. Dugourd, A., Kuppe, C., Sciacovelli, M., Gjerga, E., Gabor, A., Emdal, K. B., Vieira, V., Bekker‐Jensen, D. B., Kranz, J., Bindels, Eric. M. J., Costa, A. S. H., Sousa, A., Beltrao, P., Rocha, M., Olsen, J. V., Frezza, C., Kramann, R., &amp; Saez‐Rodriguez, J. (2021). <span class="nocase">Causal integration of multi‐omics data with prior knowledge to generate mechanistic hypotheses</span>. <em>Molecular Systems Biology</em>, <em>17</em>(1). <a href="https://doi.org/10.15252/msb.20209730">https://doi.org/10.15252/msb.20209730</a>
</div>
<div id="ref-Karr2012" class="csl-entry">
147. Karr, J. R., Sanghvi, J. C., MacKlin, D. N., Gutschow, M. V., Jacobs, J. M., Bolival, B., Assad-Garcia, N., Glass, J. I., &amp; Covert, M. W. (2012). <span class="nocase">A whole-cell computational model predicts phenotype from genotype</span>. <em>Cell</em>, <em>150</em>(2), 389–401. <a href="https://doi.org/10.1016/j.cell.2012.05.044">https://doi.org/10.1016/j.cell.2012.05.044</a>
</div>
<div id="ref-Carrera2015" class="csl-entry">
148. Carrera, J., &amp; Covert, M. W. (2015). <span>Why Build Whole-Cell Models?</span> <em>Trends in Cell Biology</em>, <em>25</em>(12), 719–722. <a href="https://doi.org/10.1016/j.tcb.2015.09.004">https://doi.org/10.1016/j.tcb.2015.09.004</a>
</div>
<div id="ref-Senft2017" class="csl-entry">
149. Senft, D., Leiserson, M. D. M., Ruppin, E., &amp; Ronai, Z. A. (2017). <span>Precision Oncology: The Road Ahead</span>. <em>Trends in Molecular Medicine</em>, <em>23</em>(10), 874–898. <a href="https://doi.org/10.1016/j.molmed.2017.08.003">https://doi.org/10.1016/j.molmed.2017.08.003</a>
</div>
<div id="ref-Fukuda2012" class="csl-entry">
150. Fukuda, K., Kobayashi, A., &amp; Watabe, K. (2012). <span class="nocase">The Role of tumor-associated macrophage in tumor progression</span>. <em>Frontiers in Bioscience - Scholar</em>, <em>2</em>, 787–798. <a href="https://doi.org/10.2741/s299">https://doi.org/10.2741/s299</a>
</div>
<div id="ref-Marku2020" class="csl-entry">
151. Marku, M., Verstraete, N., Raynal, F., Madrid-Mencía, M., Domagala, M., Fournié, J. J., Ysebaert, L., Poupot, M., &amp; Pancaldi, V. (2020). <span class="nocase">Insights on TAM formation from a boolean model of macrophage polarization based on in vitro studies</span>. <em>Cancers</em>, <em>12</em>(12), 1–23. <a href="https://doi.org/10.3390/cancers12123664">https://doi.org/10.3390/cancers12123664</a>
</div>
<div id="ref-Desoize2000" class="csl-entry">
152. Desoize, B., &amp; Jardillier, J. C. (2000). <span class="nocase">Multicellular resistance: A paradigm for clinical resistance?</span> <em>Critical Reviews in Oncology/Hematology</em>, <em>36</em>(2-3), 193–207. <a href="https://doi.org/10.1016/S1040-8428(00)00086-X">https://doi.org/10.1016/S1040-8428(00)00086-X</a>
</div>
<div id="ref-Komohara2016" class="csl-entry">
153. Komohara, Y., Fujiwara, Y., Ohnishi, K., &amp; Takeya, M. (2016). <span class="nocase">Tumor-associated macrophages: Potential therapeutic targets for anti-cancer therapy</span>. <em>Advanced Drug Delivery Reviews</em>, <em>99</em>, 180–185. <a href="https://doi.org/10.1016/j.addr.2015.11.009">https://doi.org/10.1016/j.addr.2015.11.009</a>
</div>
<div id="ref-whole-cell-doc" class="csl-entry">
154. Karr, J. R. (2020). <em><span class="nocase">An introduction to whole-cell modeling (v0.0.1)</span></em>. <a href="https://docs.karrlab.org/intro_to_wc_modeling/master/0.0.1/">https://docs.karrlab.org/intro_to_wc_modeling/master/0.0.1/</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-Tomita1999" class="csl-entry">
155. Tomita, M., Hashimoto, K., Takahashi, K., Shimizu, T. S., Matsuzaki, Y., Miyoshi, F., Saito, K., Tanida, S., Yugi, K., Venter, J. C., &amp; Hutchison, C. A. (1999). <span class="nocase">E-CELL: Software environment for whole-cell simulation</span>. <em>Bioinformatics</em>, <em>15</em>(1), 72–84. <a href="https://doi.org/10.1093/bioinformatics/15.1.72">https://doi.org/10.1093/bioinformatics/15.1.72</a>
</div>
<div id="ref-Blinov2017" class="csl-entry">
156. Blinov, M. L., Schaff, J. C., Vasilescu, D., Moraru, I. I., Bloom, J. E., &amp; Loew, L. M. (2017). <span class="nocase">Compartmental and Spatial Rule-Based Modeling with Virtual Cell</span>. <em>Biophysical Journal</em>, <em>113</em>(7), 1365–1372. <a href="https://doi.org/10.1016/j.bpj.2017.08.022">https://doi.org/10.1016/j.bpj.2017.08.022</a>
</div>
<div id="ref-Karr2015" class="csl-entry">
157. Karr, J. R., Takahashi, K., &amp; Funahashi, A. (2015). <span class="nocase">The principles of whole-cell modeling</span>. <em>Current Opinion in Microbiology</em>, <em>27</em>, 18–24. <a href="https://doi.org/10.1016/j.mib.2015.06.004">https://doi.org/10.1016/j.mib.2015.06.004</a>
</div>
<div id="ref-Macklin2020" class="csl-entry">
158. Macklin, D. N., Ahn-Horst, T. A., Choi, H., Ruggero, N. A., Carrera, J., Mason, J. C., Sun, G., Agmon, E., DeFelice, M. M., Maayan, I., Lane, K., Spangler, R. K., Gillies, T. E., Paull, M. L., Akhter, S., Bray, S. R., Weaver, D. S., Keseler, I. M., Karp, P. D., … Covert, M. W. (2020). <span class="nocase">Simultaneous cross-evaluation of heterogeneous E. coli datasets via mechanistic simulation</span>. <em>Science</em>, <em>369</em>(6502). <a href="https://doi.org/10.1126/science.aav3751">https://doi.org/10.1126/science.aav3751</a>
</div>
<div id="ref-Hallock2014" class="csl-entry">
159. Hallock, M. J., Stone, J. E., Roberts, E., Fry, C., &amp; Luthey-Schulten, Z. (2014). <span class="nocase">Simulation of reaction diffusion processes over biologically relevant size and time scales using multi-GPU workstations</span>. <em>Parallel Computing</em>, <em>40</em>(5-6), 86–99. <a href="https://doi.org/10.1016/j.parco.2014.03.009">https://doi.org/10.1016/j.parco.2014.03.009</a>
</div>
<div id="ref-HPC-wiki" class="csl-entry">
160. <em><span class="nocase">Supercomputer (Wikipedia entry)</span></em>. (2020). <a href="https://en.wikipedia.org/wiki/Supercomputer">https://en.wikipedia.org/wiki/Supercomputer</a>, (15 May 2021, date last accessed).
</div>
<div id="ref-Keating2020" class="csl-entry">
161. Keating, S. M., Waltemath, D., König, M., Zhang, F., Dräger, A., Chaouiya, C., Bergmann, F. T., Finney, A., Gillespie, C. S., Helikar, T., Hoops, S., Malik‐Sheriff, R. S., Moodie, S. L., Moraru, I. I., Myers, C. J., Naldi, A., Olivier, B. G., Sahle, S., Schaff, J. C., … Zucker, J. (2020). <span class="nocase">SBML Level 3: an extensible format for the exchange and reuse of biological models</span>. <em>Molecular Systems Biology</em>, <em>16</em>(8), 1–21. <a href="https://doi.org/10.15252/msb.20199110">https://doi.org/10.15252/msb.20199110</a>
</div>
<div id="ref-Waltemath2016" class="csl-entry">
162. Waltemath, D., Karr, J. R., Bergmann, F. T., Chelliah, V., Hucka, M., Krantz, M., Liebermeister, W., Mendes, P., Myers, C. J., Pir, P., Alaybeyoglu, B., Aranganathan, N. K., Baghalian, K., Bittig, A. T., Burke, P. E. P., Cantarelli, M., Chew, Y. H., Costa, R. S., Cursons, J., … Schreiber, F. (2016). <span class="nocase">Toward Community Standards and Software for Whole-Cell Modeling</span>. <em>IEEE Transactions on Biomedical Engineering</em>, <em>63</em>(10), 2007–2014. <a href="https://doi.org/10.1109/TBME.2016.2560762">https://doi.org/10.1109/TBME.2016.2560762</a>
</div>
<div id="ref-Burke2020" class="csl-entry">
163. Burke, P. E. P., Campos, C. B. de L., Costa, L. da F., &amp; Quiles, M. G. (2020). <span class="nocase">A biochemical network modeling of a whole-cell</span>. <em>Scientific Reports</em>, <em>10</em>(1), 13303. <a href="https://doi.org/10.1038/s41598-020-70145-4">https://doi.org/10.1038/s41598-020-70145-4</a>
</div>
<div id="ref-Carrera2014" class="csl-entry">
164. Carrera, J., Estrela, R., Luo, J., Rai, N., Tsoukalas, A., &amp; Tagkopoulos, I. (2014). <span class="nocase">An integrative, multi‐scale, genome‐wide model reveals the phenotypic landscape of Escherichia coli</span>. <em>Molecular Systems Biology</em>, <em>10</em>(7), 735. <a href="https://doi.org/10.15252/msb.20145108">https://doi.org/10.15252/msb.20145108</a>
</div>
<div id="ref-Szigeti2014" class="csl-entry">
165. Szigeti, B., Gleeson, P., Vella, M., Khayrulin, S., Palyanov, A., Hokanson, J., Currie, M., Cantarelli, M., Idili, G., &amp; Larson, S. (2014). <span class="nocase">OpenWorm: an open-science approach to modeling Caenorhabditis elegans</span>. <em>Frontiers in Computational Neuroscience</em>, <em>8</em>(November), 1–7. <a href="https://doi.org/10.3389/fncom.2014.00137">https://doi.org/10.3389/fncom.2014.00137</a>
</div>
<div id="ref-Viceconti2016" class="csl-entry">
166. Viceconti, M., &amp; Hunter, P. (2016). <span class="nocase">The Virtual Physiological Human: Ten Years after</span>. <em>Annual Review of Biomedical Engineering</em>, <em>18</em>, 103–113. <a href="https://doi.org/10.1146/annurev-bioeng-110915-114742">https://doi.org/10.1146/annurev-bioeng-110915-114742</a>
</div>
<div id="ref-Ghaffarizadeh2018" class="csl-entry">
167. Ghaffarizadeh, A., Heiland, R., Friedman, S. H., Mumenthaler, S. M., &amp; Macklin, P. (2018). <span class="nocase">PhysiCell: An open source physics-based cell simulator for 3-D multicellular systems</span>. <em>PLoS Computational Biology</em>, <em>14</em>(2). <a href="https://doi.org/10.1371/journal.pcbi.1005991">https://doi.org/10.1371/journal.pcbi.1005991</a>
</div>
<div id="ref-Letort2018" class="csl-entry">
168. Letort, G., Montagud, A., Stoll, G., Heiland, R., Barillot, E., Macklin, P., Zinovyev, A., &amp; Calzone, L. (2018). <span class="nocase">PhysiBoSS: a multi-scale agent-based modelling framework integrating physical dimension and cell signalling</span>. <em>Bioinformatics</em>, <em>35</em>(7), 1188–1196. <a href="https://doi.org/10.1093/bioinformatics/bty766">https://doi.org/10.1093/bioinformatics/bty766</a>
</div>
<div id="ref-Varela2018" class="csl-entry">
169. Varela, P. L., Ramos, C. V., Monteiro, P. T., &amp; Chaouiya, C. (2018). <span class="nocase">Epilog: A software for the logical modelling of epithelial dynamics [version 2; peer review: 3 approved]</span>. <em>F1000Research</em>, <em>7</em>, 1145. <a href="https://doi.org/10.12688/F1000RESEARCH.15613.2">https://doi.org/10.12688/F1000RESEARCH.15613.2</a>
</div>
<div id="ref-Cooper2020" class="csl-entry">
170. Cooper, F. R., Baker, R. E., Bernabeu, M. O., Bordas, R., Bowler, L., Bueno-Orovio, A., Byrne, H. M., Carapella, V., Cardone-Noott, L., Cooper, J., Dutta, S., Evans, B. D., Fletcher, A. G., Grogan, J. A., Guo, W., Harvey, D. G., Hendrix, M., Kay, D., Kursawe, J., … Gavaghan, D. J. (2020). <span class="nocase">Chaste: Cancer, Heart and Soft Tissue Environment</span>. <em>Journal of Open Source Software</em>, <em>5</em>(47), 1848. <a href="https://doi.org/10.21105/joss.01848">https://doi.org/10.21105/joss.01848</a>
</div>
<div id="ref-Stoll2020" class="csl-entry">
171. Stoll, G., Naldi, A., Noël, V., Viara, E., Barillot, E., Kroemer, G., Thieffry, D., &amp; Calzone, L. (2020). <span class="nocase">UPMaBoSS: a novel framework for dynamic cell population modeling</span>. <em>bioRxiv</em>. <a href="https://doi.org/10.1101/2020.05.31.126094">https://doi.org/10.1101/2020.05.31.126094</a>
</div>
<div id="ref-Hucka2015" class="csl-entry">
172. Hucka, M., Nickerson, D. P., Bader, G. D., Bergmann, F. T., Cooper, J., Demir, E., Garny, A., Golebiewski, M., Myers, C. J., Schreiber, F., Waltemath, D., &amp; Le Novere, N. (2015). <span class="nocase">Promoting Coordinated Development of Community-Based Information Standards for Modeling in Biology: The COMBINE Initiative</span>. <em>Frontiers in Bioengineering and Biotechnology</em>, <em>3</em>, 19. <a href="https://doi.org/10.3389/fbioe.2015.00019">https://doi.org/10.3389/fbioe.2015.00019</a>
</div>
<div id="ref-rtemps" class="csl-entry">
173. Zobolas, J. (2020). <em><span class="nocase">Rtemps: R Templates for Reproducible Data Analyses</span></em>. GitHub. <a href="https://github.com/bblodfon/rtemps">https://github.com/bblodfon/rtemps</a>
</div>
<div id="ref-Vazquez2010" class="csl-entry">
174. Vázquez, M., Nogales, R., Carmona, P., Pascual, A., &amp; Pavón, J. (2010). <em><span class="nocase">Rbbt: A Framework for Fast Bioinformatics Development with Ruby</span></em> (pp. 201–208). Springer, Berlin, Heidelberg. <a href="https://doi.org/10.1007/978-3-642-13214-8_26">https://doi.org/10.1007/978-3-642-13214-8_26</a>
</div>
</div>
</div>






















            </section>

          </div>
        </div>
      </div>
<a href="05_links.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/bblodfon/my-phd-thesis/blob/main/06_refs.Rmd",
"text": null
},
"download": [["phd-thesis.pdf", "PDF"]],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
